---
title: "16S itag analyses Fujimoto et al., submitted to Journal of Great Lakes Research"
author: "Vincent Denef"
date: "April 25, 2016"
output:
  md_document:
    toc: true
    variant: markdown_github
---
    
```{r global_options, include=FALSE}

knitr::opts_chunk$set(
  fig.width = 7, 
  fig.height = 5, 
  fig.align = 'center', 
  fig.path = 'Figures/',
  warning = FALSE, 
  message = FALSE
)

```

### Load Libraries
```{r load_libs, include=FALSE}
library(phyloseq)
library(ggplot2)
library(ape)
library(vegan)
library(plyr)
library(dplyr)
library(scales)
library(grid)
library(reshape2)
library(DESeq2)
library(pegas)
library(pgirmess)
library(multcomp)
library(multcompView)
library(grid)
library(gridExtra)
theme_set(theme_bw())

source("/Users/vdenef/Documents/UMich/Digital_Notebooks/GitHub/MicrobeMiseq/R/miseqR.R")
source("/Users/vdenef/Documents/UMich/Digital_Notebooks/GitHub/LM13_DNA/LM13_DNA_function.R")

setwd("/Users/vdenef/Documents/UMich/Digital_Notebooks/GitHub/LM13_DNA/")

### Set our theme for our plots down the road
set_ggtheme <- theme_set(theme_bw() + 
                    theme(plot.title = element_text(face="bold", size = 20),  #Set the plot title
                    strip.text.x = element_text(size=12, face="bold"),  #Set the facet titles on x-axis 
                    strip.text.y = element_text(size=12, face="bold"),  #Set the facet titles on x-axis 
                    strip.background = element_blank(),  #Set the facet background to no background
                    axis.title.x = element_text(face="bold", size=16),  #Set the x-axis title
                    axis.title.y = element_text(face="bold", size=16),  #Set the y-axis title
                    axis.text.x = element_text(colour = "black", size=14),  #Set the x-axis labels
                    axis.text.y = element_text(colour = "black", size=14),  #Set the y-axis labels
                    legend.title = element_text(size=14, face="bold"),  #Set the legend title 
                    legend.text = element_text(size = 14),  #Set the legend text
                    legend.position="right"))  #Default the legend position to the right

###set taxonomy plotting colors
group.colors <- c(Acidobacteria = "grey26", Actinobacteria = "royalblue", Alphaproteobacteria = "steelblue4", Armatimonadetes = "red", Bacteroidetes = "darkred", "BD1-5" = "chartreuse", Betaproteobacteria = "steelblue4", Caldiserica = "black","Candidate_division_BRC1" = "violetred4", "Candidate_division_JS1" = "aquamarine1", "Candidate_division_OD1" = "#6DDE88", "Candidate_division_OP3" = "hotpink", "Candidate_division_OP8" = "goldenrod1", "Candidate_division_OP11" = "chocolate4", "Candidate_division_SR1" = "tan3", "Candidate_division_TM7" = "skyblue1", "Candidate_division_WS3" = "magenta", Chlamydiae="violet", Chlorobi="cyan2", Chloroflexi="darkgreen", Cyanobacteria = "chartreuse3", Deferribacteres = "slateblue3", "Deinococcus-Thermus" = "violetred", Dictyoglomi = "cornsilk4", Deltaproteobacteria = "steelblue4", Elusimicrobia = "yellow3", Epsilonproteobacteria = "steelblue4", Fibrobacteres = "darkred", Firmicutes = "blue4", FGL7S = "palevioletred1", Fusobacteria = "slateblue1", Gammaproteobacteria = "steelblue4", Gemmatimonadetes="black", GOUTA4 = "plum1", "Hyd24-12" = "sienna2", JTB23 = "seashell2", Nitrospirae = "yellow1", "NPL-UPA2"="#652926", OC31 = "mediumpurple4", Planctomycetes = "mediumorchid3", Proteobacteria = "steelblue4", "SHA-109" = "lightsalmon3", SM2F11 = "lightskyblue2", SPOTSOCT00m83 = "orangered", Spirochaetae = "gold3", Tenericutes="pink", Thermotogae = "chocolate1", TA06 = "lightslateblue",TA18 = "rosybrown3", TM6 = "olivedrab", unclassified = "grey", Verrucomicrobia = "purple4", "WCHB1-60" = "palegreen", Other = "darkgrey", Cytophagia = "darkred", Sphingobacteria = "darkred",Flavobacteria = "darkred")

# Load physeq from workspace
#load(file = ".RData")

```

### Figure 1, panel C: Muskegon Lake buoy data   
```{r load buoy data}
#load buoy data downloaded from http://www.gvsu.edu/wri/buoy/data-generate.htm?concentration=1&date=4/23/13-11/3/13&time=0:00,12:00&x=date&y=clcon001,odo001,tp001,clcon002,tp004,clcon003,odo003,tp005
buoydata <- read.table("Input_data/wri_buoy_data.tsv",sep="\t",header=T,row.names=1,as.is=T)

#modify into dataframe with date in first column then value of parameter in second, and third colum states which parameter was measured
#also, convert fahrenheit to celcius for temps at 2m (T2), 7m (T7) and 8m (T8)
buoydata$Date <- row.names(buoydata)

T2<-data.frame(buoydata$Date,buoydata$T2)
T2$parameter<-"Temp 2m"
colnames(T2)<-c("date","value","parameter")
T2$value<-(T2$value-32)/9*5

T7<-data.frame(buoydata$Date,buoydata$T7)
T7$parameter<-"Temp 7m"
colnames(T7)<-c("date","value","parameter")
T7$value<-(T7$value-32)/9*5

T9<-data.frame(buoydata$Date,buoydata$T9)
T9$parameter<-"Temp 9m"
colnames(T9)<-c("date","value","parameter")
T9$value<-(T9$value-32)/9*5

C2<-data.frame(buoydata$Date,buoydata$CHL2)
C2$parameter<-"Chla 2m"
colnames(C2)<-c("date","value","parameter")

C5<-data.frame(buoydata$Date,buoydata$CHL5)
C5$parameter<-"Chla 5m"
colnames(C5)<-c("date","value","parameter")

DO2<-data.frame(buoydata$Date,buoydata$DO2)
DO2$parameter<-"DO 2m"
colnames(DO2)<-c("date","value","parameter")

DO8<-data.frame(buoydata$Date,buoydata$DO8)
DO8$parameter<-"DO 8m"
colnames(DO8)<-c("date","value","parameter")

buoy.df<-rbind(T2,T7,T9,C2,C5,DO2,DO8)
buoy.T<-rbind(T2,T7,T9)
buoy.C<-rbind(C2,C5)
buoy.D<-rbind(DO2,DO8)

```

```{r plot data}
buoy_plot_T <- ggplot(buoy.T,aes_string(x = "date",y = "value",group = "parameter",color = "parameter")) +
  scale_x_discrete(breaks = c("05/15/2013","07/15/2013", "08/15/2013", "09/23/2013", "11/02/2013"),labels = c("May 15", "Jul 15", "Aug 15", "Sep 23", "Nov 02"),drop = FALSE) +
  geom_line(size = 0.4) + #geom_point(size = 0.8) +
  ggtitle("") + xlab("") +
  ylab("Â°C") + 
  scale_color_manual(name = "Temp", values = c("Temp 2m" = "gray0", "Temp 7m" = "gray30", "Temp 9m" = "gray60"), breaks = c("Temp 2m", "Temp 7m" , "Temp 9m" ), labels = c("2 m","7 m","9 m")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=30, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=0),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="right", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); buoy_plot_T

buoy_plot_C <- ggplot(buoy.C,aes_string(x = "date",y = "value",group = "parameter",color = "parameter")) +
  scale_x_discrete(breaks = c("05/15/2013","07/15/2013", "08/15/2013", "09/23/2013", "11/02/2013"),labels = c("May 15", "Jul 15", "Aug 15", "Sep 23", "Nov 02"),drop = FALSE) +
  geom_line(size = 0.4) + #geom_point(size = 0.8) +
  ggtitle("") + xlab("") +
  ylab("ug/L") + 
  scale_color_manual(name = "Chla", values = c("Chla 2m" = "darkgreen", "Chla 5m" = "darkslategrey"), breaks = c("Chla 2m" , "Chla 5m"), labels = c("2 m","5 m")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=30, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=0),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="right", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); buoy_plot_C

buoy_plot_D <- ggplot(buoy.D,aes_string(x = "date",y = "value",group = "parameter",color = "parameter")) +
  scale_x_discrete(breaks = c("05/15/2013","07/15/2013", "08/15/2013", "09/23/2013", "11/02/2013"),labels = c("May 15", "Jul 15", "Aug 15", "Sep 23", "Nov 02"),drop = FALSE) +
  geom_line(size = 0.4) + #geom_point(size = 0.8) +
  ggtitle("") + xlab("") +
  ylab("mg/L") + 
  scale_color_manual(name = "DO", values = c("DO 2m" = "darkgoldenrod4", "DO 8m" = "darkgoldenrod3"), breaks = c("DO 2m", "DO 8m" ), labels = c("2 m","8 m")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=30, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="right", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); buoy_plot_D        
```

```{r final figure 1C}
pdf(file="Figures/Rplot_buoy.pdf", width= 2.5, height=3, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(buoy_plot_T,buoy_plot_C,buoy_plot_D,ncol=1)
dev.off()
```

### Import Sequence Data
```{r Import sequence Data, include=FALSE}
###input data were obtained fromn the jgi output produced by their iTagger pipeline, which performs the following steps
#(1) CONTAM FILTER: Filter one or more contaminants using Duk (e.g. PhiX control, sequencing library adapter dimers, human contaminants, etc.).  For paired reads, the entire pair is filtered if one end-read has a high-scoring hit.
#(2) PRIMER TRIM: PCR primers (of the conserved region) are trimmed away.  For paired-end reads, both forward and reverse primers must be found, otherwise the entire pair is filtered.
#(3) ITERATIVE PAIR MERGING: Reads are trimmed as a pair, removing the last base from whichever end has the highest expected error in a window 5bp wide.  Reads are trimmed from mean + 3 standard deviations to mean - 3 standard deviations in 0.5 standard deviation steps.  After each trimming step, pairs are merged into single sequences with either Flash or Pandaseq.  Pairs which are not merged continue to the next round of trimming.  Paired reads which are not combined are discarded.
#(4) EXPECTED-ERROR FILTER: Merged reads are filtered if they have an expected number of errors which exceeds the threshold.  The config file indicates the maximum number of expected errors per 100bp. Note that Flash and Pandaseq produce different quality scores in the overlap-assembled regions.

###as JGI used the wrong file to connect sample names to read files for plate 2 (DNA 16S data), we need to correct these using the following commands on the command line. The hiqualctgs is a folder available from jgi after unzipping the data files. The data folder needs to be created. the path to the script needs to be modified depending on where you place it. the script is available on the github repository and essentially changes the group names used in mothur to the correct sample names based on the conversion file which is also included on the repository.
#for file in `ls *itag`; do mkdir ../hiqualctgs/$file; done
#cd ../hiqualctgs
#for f in *itag\:; do mv "$f" "${f/itag\:/itag}";done
#cd ../data
#for file in `ls */hiqual.fastq`; do fastq_to_fasta -n -i $file |cat | awk '{if ((NR%2)==0)print prev"XXXXXX"$0;prev=$0;}' | sed 's/XXXXXX/\n/' |sed -e 's/ .*//g' -e 's/:/_/g' > ../hiqualctgs/$file; done
#cd ../hiqualctgs
#for f in `ls */*.fastq`; do t=$(echo $f | sed -e 's/\/.*$//' -e 's/_/./g' -e 's/.itag//g') && grep ">" $f |sed -e 's/>//g' -e "s/$/\t$t/g" > $t.groups; done
#cat *.groups > jgi.contigs.groups.oldnames
#rm *.groups
#cat */*.fastq > jgi.contigs.fasta
#ruby ~/software/scripts/ruby/change_name_vlookup.rb library_correction.table 0 1 hiqualctgs/jgi.contigs.groups.oldnames 1


### loading mothur output with FWDB+silva taxonomy and sample metadata. 
## create phyloseq object
sharedfile <- "Input_data/jgi.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.shared"
taxfile <- "Input_data/jgi.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.cons.taxonomy"
mothurdata <- import_mothur(mothur_shared_file = sharedfile ,mothur_constaxonomy_file = taxfile )
sampledata <- read.table("Input_data/LMSampleDescription.tsv",sep="\t",header=T,row.names=1,as.is=T)
SAMPLE = sample_data(sampledata)
physeq = merge_phyloseq(mothurdata, SAMPLE)

### We need to change the taxonomy names: when using the fwdb taxonomy we need to add different headers after removing the last column, which contains no information except for 1 Verrucomicrobia taxon
tax_table(physeq) <- tax_table(physeq)[,-8] # this removes the final column
tax_table(physeq) <- cbind(tax_table(physeq),row.names(tax_table(physeq)))
colnames(tax_table(physeq)) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Tribe","Species")

### removing non-bacterial reads (was already done in mothur, but just to be safe after merging taxonomies)
physeq <- subset_taxa(physeq, Kingdom == "Bacteria")

# Load physeq from workspace
#load(file = ".RData")

```

``` {r merge duplicates}
# Merge samples on Duplicates column
physeq_dupmerge <- merge_samples(physeq, "Duplicates")

# Weird bug, set taxa to rows
if (!taxa_are_rows(physeq_dupmerge)) {
  otu_table(physeq_dupmerge) <- t(otu_table(physeq_dupmerge))
}

## Fix metadata of merged duplicates 
# Grab unique lines (first entry) from inland phyloseq object metadata
unique_idx <- !(duplicated(sample_data(physeq)$Duplicates))
duplicate_metadata <- sample_data(physeq)[unique_idx, ]
# Assign new metadata to phyloseq object
sample_data(physeq_dupmerge) <- duplicate_metadata
physeq_dupmerge <- prune_taxa(taxa_sums(physeq_dupmerge) > 0, physeq_dupmerge)
```

```{r scaling, include=FALSE}

#check number of reads in each sample, differences in count are in part due differet numbers of chlorophyl reads depending on sample
sampsums <- data.frame(sums = sample_sums(physeq_dupmerge))

ggplot(sampsums, aes(x = sums)) + geom_histogram(binwidth = 10000)

min(sampsums)
max(sampsums)

# Scales reads to smallest library size which is 70,480
physeq.scale <- scale_reads(physeq_dupmerge, min(sample_sums(physeq_dupmerge)))
physeq.scale <- prune_taxa(taxa_sums(physeq.scale) > 0, physeq.scale)
```

```{r Proteobacteria and Bacteroidetes phylum change to class}

### ADD THE PROTEOBACTERIA CLASSES TO THE PHYLA NAME FIELD IN PHYLOSEQ OBJECT TAXONOMY 
phy <- data.frame(tax_table(physeq.scale))
Phylum <- as.character(phy$Phylum)
Class <- as.character(phy$Class)

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Proteobacteria"){
    if (Class[i] == "unclassified"){
      Phylum[i] <- Phylum[i]       
    } else {
      Phylum[i] <- Class[i]
    }
  } 
}

phy$Phylum <- Phylum
t <- tax_table(as.matrix(phy))

tax_table(physeq.scale) <- t

#same for physeq_dupmerge, but also add class name to Bacteroidetes phylum field
phy <- data.frame(tax_table(physeq_dupmerge))
Phylum <- as.character(phy$Phylum)
Class <- as.character(phy$Class)

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Proteobacteria"){
    if (Class[i] == "unclassified"){
      Phylum[i] <- Phylum[i]       
    } else {
      Phylum[i] <- Class[i]
    }
  } 
}

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Bacteroidetes"){
    if (Class[i] == "unclassified"){
      Phylum[i] <- Phylum[i] 
    } else {
      Phylum[i] <- Class[i]
    }
  } 
}

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Sphingobacteriia"){
      Phylum[i] <- "Sphingobacteria" 
    }
  } 

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Flavobacteriia"){
      Phylum[i] <- "Flavobacteria" 
    }
  } 

phy$Phylum <- Phylum
t <- tax_table(as.matrix(phy))

tax_table(physeq_dupmerge) <- t
```

##Figure 2: alpha diversity
``` {r rarefaction curves}

psdata <- physeq_dupmerge
psdata
sample_sums(psdata)

#calculate alphadiversity
set.seed(42)

#uncomment to calculate, simply reloading already calculated files
#calculate_rarefaction_curves <- function(psdata, measures, depths) {
#  require('plyr') # ldply
#  require('reshape2') # melt

#estimate_rarified_richness <- function(psdata, measures, depth) {
#    if(max(sample_sums(psdata)) < depth) return()
#    psdata <- prune_samples(sample_sums(psdata) >= depth, psdata)

#    rarified_psdata <- rarefy_even_depth(psdata, depth, verbose = FALSE)

#    alpha_diversity <- estimate_richness(rarified_psdata, measures = measures)

    # as.matrix forces the use of melt.array, which includes the Sample names (rownames)
#    molten_alpha_diversity <- melt(as.matrix(alpha_diversity), varnames = c('Sample', 'Measure'), value.name = 'Alpha_diversity')

#    molten_alpha_diversity
#  }

#  names(depths) <- depths # this enables automatic addition of the Depth to the output by ldply
#  rarefaction_curve_data <- ldply(depths, estimate_rarified_richness, psdata = psdata, measures = measures, .id = 'Depth', .progress = ifelse(interactive(), 'text', 'none'))

  # convert Depth from factor to numeric
#  rarefaction_curve_data$Depth <- as.numeric(levels(rarefaction_curve_data$Depth))[rarefaction_curve_data$Depth]

#  rarefaction_curve_data
#}

#rarefaction_curve_data <- calculate_rarefaction_curves(psdata, c('Observed'), rep(c(1, 10, 100, 1000, 1:100 * 10000), #each = 10))
#summary(rarefaction_curve_data)

#summarize alpha div
#rarefaction_curve_data_summary <- ddply(rarefaction_curve_data, c('Depth', 'Sample', 'Measure'), summarise, Alpha_diversity_mean = mean(Alpha_diversity), Alpha_diversity_sd = sd(Alpha_diversity))

# add sample data
#rarefaction_curve_data_summary_verbose <- merge(rarefaction_curve_data_summary, data.frame(sample_data(psdata)), by.x = 'Sample', by.y = 'row.names')
#for(i in 1:length(rarefaction_curve_data_summary_verbose$Season)){
#  rarefaction_curve_data_summary_verbose$Depth_Seas[i]<-paste(as.character(rarefaction_curve_data_summary_verbose$Depth_C[i]),
#                                  as.character(rarefaction_curve_data_summary_verbose$Season[i]))}

#write.table(rarefaction_curve_data_summary_verbose, "Input_data/rarefaction_curve_data_summary_verbose", row.names = TRUE)

rarefaction_curve_data_summary_verbose<-read.table("Input_data/rarefaction_curve_data_summary_verbose", header = TRUE)

#plot
rar.all <- ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    color = Season, shape = Station, size = Fraction, fill = Depth_Seas,
    group = Sample)) + 
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  geom_vline(xintercept = 70480, colour="grey", linetype = "longdash") +
  geom_line() + geom_pointrange(alpha=0.5) + 
  facet_wrap(
  facets = ~ Measure,
  scales = 'free_y') +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rar.all


#plot
rar.station <- ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    color = Station,
    group = Sample)) + 
  scale_color_manual(name = "station", breaks = c("MLB","MM15","MM110"), 
                     labels = c("MLB","M15","M110"),
                     values = c("MLB" = "darkgreen", "MM15" = "green", "MM110" = "blue")) +
  geom_vline(xintercept = 70480, colour="grey", linetype = "longdash") +
  geom_line() + geom_pointrange() + 
  facet_wrap(
  facets = ~ Measure,
  scales = 'free_y') +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rar.station

rar.season <- ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    color = Season,
    group = Sample)) + 
    scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  geom_vline(xintercept = 70480, colour="grey", linetype = "longdash") +
  geom_line() + geom_pointrange() + 
  facet_wrap(
  facets = ~ Measure,
  scales = 'free_y') +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rar.season

rar.fraction <- ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    color = Fraction,
    group = Sample)) + 
    scale_color_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = "grey", "E" = "black")) +
  geom_vline(xintercept = 70480, colour="grey", linetype = "longdash") +
  geom_line() + geom_pointrange() + 
  facet_wrap(
  facets = ~ Measure,
  scales = 'free_y') +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rar.fraction

rar.depth <- ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    fill = Depth_C,
    group = Sample)) + 
    scale_fill_manual(name = "depth", breaks = c("D", "M", "S"), 
                     labels = c("Deep","Chla max","Surface"),
                     values = c("D" = "black", "M" = "grey", "S" = "white")) +
  geom_vline(xintercept = 70480, colour="grey", linetype = "longdash") +
  geom_line() + geom_pointrange(shape=21) + 
  facet_wrap(
  facets = ~ Measure,
  scales = 'free_y') +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rar.depth
```

```{r final figure S1: rarefaction}
pdf(file="Figures/Rplot_rarefaction.pdf", width= 7, height=7, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(rar.fraction,rar.season,rar.station,rar.depth,ncol=2)
dev.off()
```

```{r alpha diversity metrics calculation}
# For alpha diversity we will RAREFY our phyloseq object.  Our raw-data phyloseq object was:
physeq_dupmerge  # Raw Merged Samples
min(sample_sums(physeq_dupmerge))


# Following code from Michelle Berry's Butterflygut website: http://rstudio-pubs-static.s3.amazonaws.com/25722_9fc9bdc48f0d4f13b05fa61baeda57a0.html#alpha-diversity
# Rarefy to 70480 reads with replacement 100 times to estimate species richness
# Initialize matrices to store richness and evenness estimates
#richness <- matrix(nrow = 47,ncol = 100)  # Store richness:  We have 47 samples 
#row.names(richness) <- sample_names(physeq_dupmerge)
#evenness <- matrix(nrow = 47,ncol = 100)  #Store evenness
#row.names(evenness) <- sample_names(physeq_dupmerge)

# We want to be reproducible - so let's set the seed.
set.seed(3)

# For 100 replications, rarefy the OTU table to 70480 reads and store the richness and evennes estimates in our 2 matrices we just created. Run this once, afterwards load saved data
#for (i in 1:100) {
# r <- rarefy_even_depth(physeq_dupmerge, sample.size = 70480, verbose = FALSE, replace = TRUE)
#  rich <- as.numeric(as.matrix(estimate_richness(r, measures="Observed")))
#  richness[,i] <- rich
#  even <- as.numeric(as.matrix(estimate_richness(r, measures="InvSimpson")))
#  evenness[,i] <- even
#}

#write.table(richness, "Input_data/ObservedRichness100", row.names = TRUE)
#write.table(evenness, "Input_data/InvSimpson100", row.names = TRUE)

richness <- read.table("Input_data/ObservedRichness100", header = TRUE)
evenness <- read.table("Input_data/InvSimpson100", header = TRUE)

# Create a new matrix to hold the means and standard deviations of all the richness estimates
rich_stats = matrix(nrow= nrow(richness), ncol=1)
rich_stats[,1] = apply(richness, 1, mean)
#rich_stats[,2] = apply(richness, 1, sd)
rich_stats = data.frame(row.names(richness), rich_stats)
colnames(rich_stats) = c("samples","mean")
rich_stats$Test <- "Observed Richness"
rich_stats$Duplicates <- as.character(rich_stats$samples)
rich_stats <- merge(rich_stats,data.frame(duplicate_metadata),by="Duplicates")
for(i in 1:length(rich_stats$Season)){
  rich_stats$Depth_Seas[i]<-paste(as.character(rich_stats$Depth_C[i]),
                                  as.character(rich_stats$Season[i]))}
for(i in 1:length(rich_stats$Season)){
  rich_stats$Stat_Seas[i]<-paste(as.character(rich_stats$Station[i]),
                                  as.character(rich_stats$Season[i]))}
for(i in 1:length(rich_stats$Season)){
  rich_stats$Frac_Seas[i]<-paste(as.character(rich_stats$Fraction[i]),
                                  as.character(rich_stats$Season[i]))}

# Create a new matrix to hold the means and standard deviations of the evenness estimates
even_stats = matrix(nrow = nrow(evenness), ncol = 1)
even_stats[,1] = apply(evenness, 1, mean)
#even_stats[,2] = apply(evenness, 1, sd)
even_stats = data.frame(row.names(evenness), even_stats)
colnames(even_stats) = c("samples","mean")
even_stats$Test <- "Inverse Simpson"
even_stats$Duplicates <- as.character(even_stats$samples)
even_stats <- merge(even_stats,data.frame(duplicate_metadata),by="Duplicates")

###  Add SIMPSON'S EVENNESS!
#simps_even = as.data.frame(matrix(nrow = nrow(evenness), ncol = 2))
simps_even <- data.frame(matrix(nrow = nrow(evenness), ncol = 3))
colnames(simps_even) = c("samples","mean", "Test")
simps_even$samples <- even_stats$samples
simps_even$mean <-  even_stats$mean/rich_stats$mean
simps_even$Test <- "Simpson's Evenness"
simps_even$Duplicates <- as.character(simps_even$samples)
simps_even <- merge(simps_even,data.frame(duplicate_metadata),by="Duplicates")
for(i in 1:length(simps_even$Season)){
  simps_even$Depth_Seas[i]<-paste(as.character(simps_even$Depth_C[i]),
                                  as.character(simps_even$Season[i]))}
for(i in 1:length(simps_even$Season)){
  simps_even$Stat_Seas[i]<-paste(as.character(simps_even$Station[i]),
                                  as.character(simps_even$Season[i]))}
for(i in 1:length(simps_even$Season)){
  simps_even$Frac_Seas[i]<-paste(as.character(simps_even$Fraction[i]),
                                  as.character(simps_even$Season[i]))}
```

```{r richness plots}
###RICHNESS PLOTS
#adjust default color of boxplot outliers so that it matches boxplot color
update_geom_defaults("point", list(colour = NULL))
## 1. richness differences based on Season
rich_Season_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Season)) 
print(rich_Season_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Season_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Season))  ## Defaults to P < 0.05
print(rich_Season_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Season_KW_test <- rich_Season_KW_MC$dif.com$difference # select logical vector
names(rich_Season_KW_test) <- row.names(rich_Season_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Season_KW_letters <- multcompLetters(rich_Season_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Season_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Season_KW_letters$Letters)), as.vector(rich_Season_KW_letters$Letters))
colnames(rich_Season_KW_sigs_dataframe) <- c("Season", "siglabel")
rich_Season_KW_try <- merge(rich_stats, rich_Season_KW_sigs_dataframe, by = "Season")

rich_Season_KW_try$Season <-factor(rich_Season_KW_try$Season,levels=c( "Sp", "Su", "Fa"))

rich_Season_KW_sigs <- rich_Season_KW_try %>%
  select(Season, Season, siglabel) %>%
  distinct()

# finally, the plot, after first setting the order of the factor levels
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Season_KW_try$Fraction <- factor(rich_Season_KW_try$Fraction,levels=c("B","E"))
rich_Season_KW_try$Season <- factor(rich_Season_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Season_KW_try$Station <- factor(rich_Season_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Season_KW_try$Depth_C <- factor(rich_Season_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Season_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Season_boxplot <- ggplot(rich_Season_KW_try, aes(x = Season, y = mean))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("Season") +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = rich_Season_KW_sigs, aes(label = siglabel, x = Season, y = .95), size = 4) +
  scale_color_manual(name = "season", values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),
                     labels = c("Spring",  "Summer", "Fall")) +

#  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
#                     labels = c("Estuary", "Near",  "Off")) + 
#  scale_size_manual(name = "fraction", breaks = c("B","E"), 
#                     labels = c("Free","Particle"),
#                     values = c("B" = 1, "E" = 2)) +
#    scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
#                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
#                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("Sp", "Su", "Fa"),
                   labels=c("Spring\n(n=14)","Summer\n(n=17)", "Fall\n(n=16)")) + 
  xlab("") + ylab("Observed Richness") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Season_boxplot #top, right, bottom, left
dev.off()

## 2. richness differences based on Fraction
rich_Fraction_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Fraction)) 
print(rich_Fraction_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Fraction_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Fraction))  ## Defaults to P < 0.05
print(rich_Fraction_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Fraction_KW_test <- rich_Fraction_KW_MC$dif.com$difference # select logical vector
names(rich_Fraction_KW_test) <- row.names(rich_Fraction_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Fraction_KW_letters <- multcompLetters(rich_Fraction_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Fraction_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Fraction_KW_letters$Letters)), as.vector(rich_Fraction_KW_letters$Letters))
colnames(rich_Fraction_KW_sigs_dataframe) <- c("Fraction", "siglabel")
rich_Fraction_KW_try <- merge(rich_stats, rich_Fraction_KW_sigs_dataframe, by = "Fraction")

rich_Fraction_KW_try$Fraction <-factor(rich_Fraction_KW_try$Fraction,levels=c( "B", "E"))

rich_Fraction_KW_sigs <- rich_Fraction_KW_try %>%
  select(Fraction, Fraction, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Fraction_KW_try$Fraction <- factor(rich_Fraction_KW_try$Fraction,levels=c("B","E"))
rich_Fraction_KW_try$Season <- factor(rich_Fraction_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Fraction_KW_try$Station <- factor(rich_Fraction_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Fraction_KW_try$Depth_C <- factor(rich_Fraction_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Fraction_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Fraction_boxplot <- ggplot(rich_Fraction_KW_try, aes(x = Fraction, y = mean))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("Fraction") +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = rich_Fraction_KW_sigs, aes(label = siglabel, x = Fraction, y = .95), size = 4) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
 scale_x_discrete(breaks=c("B", "E"),
                   labels=c("FL\n(n=24)","PA\n(n=23)")) + 
   xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Fraction_boxplot #top, right, bottom, left
dev.off()


## 3. richness differences based on Station
rich_Station_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Station)) 
print(rich_Station_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Station_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Station))  ## Defaults to P < 0.05
print(rich_Station_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Station_KW_test <- rich_Station_KW_MC$dif.com$difference # select logical vector
names(rich_Station_KW_test) <- row.names(rich_Station_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Station_KW_letters <- multcompLetters(rich_Station_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Station_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Station_KW_letters$Letters)), as.vector(rich_Station_KW_letters$Letters))
colnames(rich_Station_KW_sigs_dataframe) <- c("Station", "siglabel")
rich_Station_KW_try <- merge(rich_stats, rich_Station_KW_sigs_dataframe, by = "Station")

rich_Station_KW_try$Station <-factor(rich_Station_KW_try$Station,levels=c( "MLB", "MM15", "MM110"))

rich_Station_KW_sigs <- rich_Station_KW_try %>%
  select(Station, Station, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Station_KW_try$Fraction <- factor(rich_Station_KW_try$Fraction,levels=c("B","E"))
rich_Station_KW_try$Season <- factor(rich_Station_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Station_KW_try$Station <- factor(rich_Station_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Station_KW_try$Depth_C <- factor(rich_Station_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Station_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Station_boxplot <- ggplot(rich_Station_KW_try, aes(x = Station, y = mean))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("Station") +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = rich_Station_KW_sigs, aes(label = siglabel, x = Station, y = .95), size = 4) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_x_discrete(breaks=c("MLB", "MM15", "MM110"),
                   labels=c("Estuary\n(n=10)","Near\n(n=18)", "Off\n(n=19)")) + 
  xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Station_boxplot #top, right, bottom, left
dev.off()


## 4. richness differences based on Depth_C
#rich_stats_off <- rich_stats[rich_stats$Station == "MM110" & rich_stats$Season == "Su", ]

rich_Depth_C_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Depth_C)) 
print(rich_Depth_C_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Depth_C_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Depth_C))  ## Defaults to P < 0.05
print(rich_Depth_C_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Depth_C_KW_test <- rich_Depth_C_KW_MC$dif.com$difference # select logical vector
names(rich_Depth_C_KW_test) <- row.names(rich_Depth_C_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Depth_C_KW_letters <- multcompLetters(rich_Depth_C_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Depth_C_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Depth_C_KW_letters$Letters)), as.vector(rich_Depth_C_KW_letters$Letters))
colnames(rich_Depth_C_KW_sigs_dataframe) <- c("Depth_C", "siglabel")
rich_Depth_C_KW_try <- merge(rich_stats, rich_Depth_C_KW_sigs_dataframe, by = "Depth_C")

rich_Depth_C_KW_try$Depth_C <-factor(rich_Depth_C_KW_try$Depth_C,levels=c( "S", "M", "D"))

rich_Depth_C_KW_sigs <- rich_Depth_C_KW_try %>%
  select(Depth_C, Depth_C, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Depth_C_KW_try$Fraction <- factor(rich_Depth_C_KW_try$Fraction,levels=c("B","E"))
rich_Depth_C_KW_try$Season <- factor(rich_Depth_C_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Depth_C_KW_try$Station <- factor(rich_Depth_C_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Depth_C_KW_try$Depth_C <- factor(rich_Depth_C_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Depth_C_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Depth_C_boxplot <- ggplot(rich_Depth_C_KW_try, aes(x = Depth_C, y = mean))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("Depth") +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = rich_Depth_C_KW_sigs, aes(label = siglabel, x = Depth_C, y = .95), size = 4) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_x_discrete(breaks=c("S", "M", "D"),
                   labels=c("Surface\n(n=30)","Chla max\n(n=2)","Deep\n(n=15)")) + 
  xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Depth_C_boxplot #top, right, bottom, left
dev.off()

```

```{r richness plot combination of factors to create multicomp letters based on Kruskall Wallis test between sub boxplots data}
###RICHNESS PLOTS
## 1. richness differences based on Season and Station combination
rich_Stat_Seas_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Stat_Seas)) 
print(rich_Stat_Seas_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Stat_Seas_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Stat_Seas))  ## Defaults to P < 0.05
print(rich_Stat_Seas_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Stat_Seas_KW_test <- rich_Stat_Seas_KW_MC$dif.com$difference # select logical vector
names(rich_Stat_Seas_KW_test) <- row.names(rich_Stat_Seas_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Stat_Seas_KW_letters <- multcompLetters(rich_Stat_Seas_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Stat_Seas_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Stat_Seas_KW_letters$Letters)), as.vector(rich_Stat_Seas_KW_letters$Letters))
colnames(rich_Stat_Seas_KW_sigs_dataframe) <- c("Stat_Seas", "siglabel")
rich_Stat_Seas_KW_try <- merge(rich_stats, rich_Stat_Seas_KW_sigs_dataframe, by = "Stat_Seas")

rich_Stat_Seas_KW_try$Stat_Seas <-factor(rich_Stat_Seas_KW_try$Stat_Seas,levels=c( "MLB Sp","MM15 Sp","MM110 Sp","MLB Su","MM15 Su","MM110 Su","MLB Fa","MM15 Fa","MM110 Fa"))

rich_Stat_Seas_KW_sigs <- rich_Stat_Seas_KW_try %>%
  select(Stat_Seas, Stat_Seas, siglabel) %>%
  distinct()

# finally, the plot, after first setting the order of the factor levels
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Stat_Seas_KW_try$Fraction <- factor(rich_Stat_Seas_KW_try$Fraction,levels=c("B","E"))
rich_Stat_Seas_KW_try$Season <- factor(rich_Stat_Seas_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Stat_Seas_KW_try$Station <- factor(rich_Stat_Seas_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Stat_Seas_KW_try$Depth_C <- factor(rich_Stat_Seas_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Seas_Stat_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Seas_Stat_boxplot <- ggplot(rich_Stat_Seas_KW_try, aes(x = Stat_Seas, y = mean))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("Season & Station") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = rich_Stat_Seas_KW_sigs, aes(label = siglabel, x = Stat_Seas, y = .95), size = 2) +
  scale_color_manual(name = "season", values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),
                     labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                     labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
    scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa"),
                   labels=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa")) + 
  xlab("") + ylab("Observed Richness") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Seas_Stat_boxplot #top, right, bottom, left
dev.off()

## 2. richness differences based on Fraction and Season
rich_Frac_Seas_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Frac_Seas)) 
print(rich_Frac_Seas_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Frac_Seas_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Frac_Seas))  ## Defaults to P < 0.05
print(rich_Frac_Seas_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Frac_Seas_KW_test <- rich_Frac_Seas_KW_MC$dif.com$difference # select logical vector
names(rich_Frac_Seas_KW_test) <- row.names(rich_Frac_Seas_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Frac_Seas_KW_letters <- multcompLetters(rich_Frac_Seas_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Frac_Seas_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Frac_Seas_KW_letters$Letters)), as.vector(rich_Frac_Seas_KW_letters$Letters))
colnames(rich_Frac_Seas_KW_sigs_dataframe) <- c("Frac_Seas", "siglabel")
rich_Frac_Seas_KW_try <- merge(rich_stats, rich_Frac_Seas_KW_sigs_dataframe, by = "Frac_Seas")

rich_Frac_Seas_KW_try$Frac_Seas <-factor(rich_Frac_Seas_KW_try$Frac_Seas,levels=c( "B Sp", "B Su", "B Fa", "E Sp", "E Su", "E Fa"))

rich_Frac_Seas_KW_sigs <- rich_Frac_Seas_KW_try %>%
  select(Frac_Seas, Frac_Seas, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Frac_Seas_KW_try$Fraction <- factor(rich_Frac_Seas_KW_try$Fraction,levels=c("B","E"))
rich_Frac_Seas_KW_try$Season <- factor(rich_Frac_Seas_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Frac_Seas_KW_try$Station <- factor(rich_Frac_Seas_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Frac_Seas_KW_try$Depth_C <- factor(rich_Frac_Seas_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Frac_Seas_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Frac_Seas_boxplot <- ggplot(rich_Frac_Seas_KW_try, aes(x = Frac_Seas, y = mean))  +
  geom_boxplot(aes(color=Season),outlier.size = 0)  + ggtitle("Fraction & Season") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = rich_Frac_Seas_KW_sigs, aes(label = siglabel, x = Frac_Seas, y = .95), size = 2) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                      labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
 scale_x_discrete(breaks=c("B Sp", "B Su", "B Fa", "E Sp", "E Su", "E Fa"),
                   labels=c("FL Sp", "FL Su", "FL Fa", "PA Sp", "PA Su", "PA Fa")) + 
   xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Frac_Seas_boxplot #top, right, bottom, left
dev.off()


## 3. richness differences based on Station and Season
#  Resort the factors of Season & Fraction object to have primary sorting by station
rich_Stat_Seas_KW_try$Stat_Seas <-factor(rich_Stat_Seas_KW_try$Stat_Seas,levels=c( "MLB Sp","MLB Su","MLB Fa","MM15 Sp","MM15 Su","MM15 Fa","MM110 Sp","MM110 Su","MM110 Fa"))

rich_Stat_Seas_KW_sigs <- rich_Stat_Seas_KW_try %>%
  select(Stat_Seas, Stat_Seas, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Station_KW_try$Fraction <- factor(rich_Station_KW_try$Fraction,levels=c("B","E"))
rich_Station_KW_try$Season <- factor(rich_Station_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Station_KW_try$Station <- factor(rich_Station_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Station_KW_try$Depth_C <- factor(rich_Station_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Stat_Seas_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Stat_Seas_boxplot <- ggplot(rich_Stat_Seas_KW_try, aes(x = Stat_Seas, y = mean))  +
  geom_boxplot(aes(color=Season),outlier.size = 0)  + ggtitle("Station & Season") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = rich_Stat_Seas_KW_sigs, aes(label = siglabel, x = Stat_Seas, y = .95), size = 2) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                      labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa"),
                   labels=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa")) +   xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Stat_Seas_boxplot #top, right, bottom, left
dev.off()


## 4. richness differences based on Depth_C and Season
#rich_stats_off <- rich_stats[rich_stats$Station == "MM110" & rich_stats$Season == "Su", ]

rich_Depth_Seas_KW <- kruskal.test(rich_stats$mean ~ as.factor(rich_stats$Depth_Seas)) 
print(rich_Depth_Seas_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
rich_Depth_Seas_KW_MC <- kruskalmc(rich_stats$mean ~ as.factor(rich_stats$Depth_Seas))  ## Defaults to P < 0.05
print(rich_Depth_Seas_KW_MC)

# Test to find the letters to represent significance in a plot
rich_Depth_Seas_KW_test <- rich_Depth_Seas_KW_MC$dif.com$difference # select logical vector
names(rich_Depth_Seas_KW_test) <- row.names(rich_Depth_Seas_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
rich_Depth_Seas_KW_letters <- multcompLetters(rich_Depth_Seas_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
rich_Depth_Seas_KW_sigs_dataframe <-  data.frame(as.vector(names(rich_Depth_Seas_KW_letters$Letters)), as.vector(rich_Depth_Seas_KW_letters$Letters))
colnames(rich_Depth_Seas_KW_sigs_dataframe) <- c("Depth_Seas", "siglabel")
rich_Depth_Seas_KW_try <- merge(rich_stats, rich_Depth_Seas_KW_sigs_dataframe, by = "Depth_Seas")

rich_Depth_Seas_KW_try$Depth_Seas <-factor(rich_Depth_Seas_KW_try$Depth_Seas,levels=c( "S Sp", "S Su", "S Fa", "M Su", "D Sp", "D Su", "D Fa"))

rich_Depth_Seas_KW_sigs <- rich_Depth_Seas_KW_try %>%
  select(Depth_Seas, Depth_Seas, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Richness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
rich_Depth_Seas_KW_try$Fraction <- factor(rich_Depth_Seas_KW_try$Fraction,levels=c("B","E"))
rich_Depth_Seas_KW_try$Season <- factor(rich_Depth_Seas_KW_try$Season,levels=c("Sp","Su","Fa"))
rich_Depth_Seas_KW_try$Station <- factor(rich_Depth_Seas_KW_try$Station,levels=c("MLB","MM15","MM110"))
rich_Depth_Seas_KW_try$Depth_C <- factor(rich_Depth_Seas_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Richness_Depth_Seas_boxplot.pdf", width= 5, height=5, pointsize= 8)
rich_Depth_Seas_boxplot <- ggplot(rich_Depth_Seas_KW_try, aes(x = Depth_Seas, y = mean))  +
  geom_boxplot(aes(color=Season),outlier.size = 0)  + ggtitle("Depth & Season") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = rich_Depth_Seas_KW_sigs, aes(label = siglabel, x = Depth_Seas, y = .95), size = 2) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                      labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("S Sp", "S Su", "S Fa", "M Su", "D Sp", "D Su", "D Fa"),
                   labels=c("S Sp", "S Su", "S Fa", "M Su", "D Sp", "D Su", "D Fa")) + 
  xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); rich_Depth_Seas_boxplot #top, right, bottom, left
dev.off()
```


```{r evenness plots} 
###EVENNESS PLOTS
## 1. evenness differences based on Season
even_Season_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Season)) 
print(even_Season_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Season_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Season))  ## Defaults to P < 0.05
print(even_Season_KW_MC)

# Test to find the letters to represent significance in a plot
even_Season_KW_test <- even_Season_KW_MC$dif.com$difference # select logical vector
names(even_Season_KW_test) <- row.names(even_Season_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Season_KW_letters <- multcompLetters(even_Season_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Season_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Season_KW_letters$Letters)), as.vector(even_Season_KW_letters$Letters))
colnames(even_Season_KW_sigs_dataframe) <- c("Season", "siglabel")
even_Season_KW_try <- merge(simps_even, even_Season_KW_sigs_dataframe, by = "Season")

even_Season_KW_try$Season <-factor(even_Season_KW_try$Season,levels=c( "Sp", "Su", "Fa"))

even_Season_KW_sigs <- even_Season_KW_try %>%
  select(Season, Season, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Season_KW_try$Fraction <- factor(even_Season_KW_try$Fraction,levels=c("B","E"))
even_Season_KW_try$Season <- factor(even_Season_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Season_KW_try$Station <- factor(even_Season_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Season_KW_try$Depth_C <- factor(even_Season_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Evenness_Season_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Season_boxplot <- ggplot(even_Season_KW_try, aes(x = Season, y = mean))  +
  geom_boxplot(outlier.size = 0)  + #ggtitle(alpha_title) +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = even_Season_KW_sigs, aes(label = siglabel, x = Season, y = 0), size = 4) +
  scale_color_manual(name = "season", values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),
                     labels = c("Spring",  "Summer", "Fall")) +
  scale_x_discrete(breaks=c("Sp", "Su", "Fa"),
                   labels=c("Spring\n(n=14)","Summer\n(n=17)", "Fall\n(n=16)")) + 
  xlab("") + ylab("Simpson's Evenness") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Season_boxplot #top, right, bottom, left
dev.off()


## 2. Evenness differences based on Fraction
even_Fraction_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Fraction)) 
print(even_Fraction_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Fraction_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Fraction))  ## Defaults to P < 0.05
print(even_Fraction_KW_MC)

# Test to find the letters to represent significance in a plot
even_Fraction_KW_test <- even_Fraction_KW_MC$dif.com$difference # select logical vector
names(even_Fraction_KW_test) <- row.names(even_Fraction_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Fraction_KW_letters <- multcompLetters(even_Fraction_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Fraction_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Fraction_KW_letters$Letters)), as.vector(even_Fraction_KW_letters$Letters))
colnames(even_Fraction_KW_sigs_dataframe) <- c("Fraction", "siglabel")
even_Fraction_KW_try <- merge(simps_even, even_Fraction_KW_sigs_dataframe, by = "Fraction")

even_Fraction_KW_try$Fraction <-factor(even_Fraction_KW_try$Fraction,levels=c( "B", "E"))

even_Fraction_KW_sigs <- even_Fraction_KW_try %>%
  select(Fraction, Fraction, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Fraction_KW_try$Fraction <- factor(even_Fraction_KW_try$Fraction,levels=c("B","E"))
even_Fraction_KW_try$Season <- factor(even_Fraction_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Fraction_KW_try$Station <- factor(even_Fraction_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Fraction_KW_try$Depth_C <- factor(even_Fraction_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Evenness_Fraction_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Fraction_boxplot <- ggplot(even_Fraction_KW_try, aes(x = Fraction, y = mean))  +
  geom_boxplot(outlier.size = 0)  + #ggtitle(alpha_title) +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = even_Fraction_KW_sigs, aes(label = siglabel, x = Fraction, y = 0), size = 4) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
 scale_x_discrete(breaks=c("B", "E"),
                   labels=c("FL\n(n=24)","PA\n(n=23)")) + 
   xlab("") + ylab("") + 
  theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Fraction_boxplot #top, right, bottom, left
dev.off()

## 3. Evenness differences based on Station
even_Station_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Station)) 
print(even_Station_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Station_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Station))  ## Defaults to P < 0.05
print(even_Station_KW_MC)

# Test to find the letters to represent significance in a plot
even_Station_KW_test <- even_Station_KW_MC$dif.com$difference # select logical vector
names(even_Station_KW_test) <- row.names(even_Station_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Station_KW_letters <- multcompLetters(even_Station_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Station_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Station_KW_letters$Letters)), as.vector(even_Station_KW_letters$Letters))
colnames(even_Station_KW_sigs_dataframe) <- c("Station", "siglabel")
even_Station_KW_try <- merge(simps_even, even_Station_KW_sigs_dataframe, by = "Station")

even_Station_KW_try$Station <-factor(even_Station_KW_try$Station,levels=c( "MLB", "MM15", "MM110"))

even_Station_KW_sigs <- even_Station_KW_try %>%
  select(Station, Station, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Station_KW_try$Fraction <- factor(even_Station_KW_try$Fraction,levels=c("B","E"))
even_Station_KW_try$Season <- factor(even_Station_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Station_KW_try$Station <- factor(even_Station_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Station_KW_try$Depth_C <- factor(even_Station_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Evenness_Station_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Station_boxplot <- ggplot(even_Station_KW_try, aes(x = Station, y = mean))  +
  geom_boxplot(outlier.size = 0)  + #ggtitle(alpha_title) +
geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = even_Station_KW_sigs, aes(label = siglabel, x = Station, y = 0), size = 4) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
 scale_x_discrete(breaks=c("MLB", "MM15", "MM110"),
                   labels=c("Estuary\n(n=10)","Near\n(n=18)", "Off\n(n=19)")) + 
  xlab("") + ylab("") + 
  theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Station_boxplot #top, right, bottom, left
dev.off()

## 4. Evenness differences based on Depth_C
#simps_even_off <- simps_even[simps_even$Station == "MM110" & simps_even$Season == "Su", ]

even_Depth_C_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Depth_C)) 
print(even_Depth_C_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Depth_C_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Depth_C))  ## Defaults to P < 0.05
print(even_Depth_C_KW_MC)

# Test to find the letters to represent significance in a plot
even_Depth_C_KW_test <- even_Depth_C_KW_MC$dif.com$difference # select logical vector
names(even_Depth_C_KW_test) <- row.names(even_Depth_C_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Depth_C_KW_letters <- multcompLetters(even_Depth_C_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Depth_C_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Depth_C_KW_letters$Letters)), as.vector(even_Depth_C_KW_letters$Letters))
colnames(even_Depth_C_KW_sigs_dataframe) <- c("Depth_C", "siglabel")
even_Depth_C_KW_try <- merge(simps_even, even_Depth_C_KW_sigs_dataframe, by = "Depth_C")

even_Depth_C_KW_try$Depth_C <-factor(even_Depth_C_KW_try$Depth_C,levels=c( "S", "M", "D"))

even_Depth_C_KW_sigs <- even_Depth_C_KW_try %>%
  select(Depth_C, Depth_C, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("Evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Depth_C_KW_try$Fraction <- factor(even_Depth_C_KW_try$Fraction,levels=c("B","E"))
even_Depth_C_KW_try$Season <- factor(even_Depth_C_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Depth_C_KW_try$Station <- factor(even_Depth_C_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Depth_C_KW_try$Depth_C <- factor(even_Depth_C_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_Evenness_Depth_C_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Depth_C_boxplot <- ggplot(even_Depth_C_KW_try, aes(x = Depth_C, y = mean))  +
  geom_boxplot(outlier.size = 0)  + #ggtitle(alpha_title) +
  geom_point(aes(color=Season),alpha=0.5, position = position_dodge(0.8)) +
  geom_text(data = even_Depth_C_KW_sigs, aes(label = siglabel, x = Depth_C, y = 0), size = 4) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_x_discrete(breaks=c("S", "M", "D"),
                   labels=c("Surface\n(n=30)","Chla max\n(n=2)","Deep\n(n=15)")) + 
  xlab("") + ylab("") + 
  theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Depth_C_boxplot #top, right, bottom, left
dev.off()

```

```{r evenness plots combination of factors to create multicomp letters based on Kruskall Wallis test between sub boxplots data}
## 1. evenness differences based on Season and Station combination
even_Stat_Seas_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Stat_Seas)) 
print(even_Stat_Seas_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Stat_Seas_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Stat_Seas))  ## Defaults to P < 0.05
print(even_Stat_Seas_KW_MC)

# Test to find the letters to represent significance in a plot
even_Stat_Seas_KW_test <- even_Stat_Seas_KW_MC$dif.com$difference # select logical vector
names(even_Stat_Seas_KW_test) <- row.names(even_Stat_Seas_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Stat_Seas_KW_letters <- multcompLetters(even_Stat_Seas_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Stat_Seas_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Stat_Seas_KW_letters$Letters)), as.vector(even_Stat_Seas_KW_letters$Letters))
colnames(even_Stat_Seas_KW_sigs_dataframe) <- c("Stat_Seas", "siglabel")
even_Stat_Seas_KW_try <- merge(simps_even, even_Stat_Seas_KW_sigs_dataframe, by = "Stat_Seas")

even_Stat_Seas_KW_try$Stat_Seas <-factor(even_Stat_Seas_KW_try$Stat_Seas,levels=c( "MLB Sp","MM15 Sp","MM110 Sp","MLB Su","MM15 Su","MM110 Su","MLB Fa","MM15 Fa","MM110 Fa"))

even_Stat_Seas_KW_sigs <- even_Stat_Seas_KW_try %>%
  select(Stat_Seas, Stat_Seas, siglabel) %>%
  distinct()

# finally, the plot, after first setting the order of the factor levels
alpha_title <- paste("evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Stat_Seas_KW_try$Fraction <- factor(even_Stat_Seas_KW_try$Fraction,levels=c("B","E"))
even_Stat_Seas_KW_try$Season <- factor(even_Stat_Seas_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Stat_Seas_KW_try$Station <- factor(even_Stat_Seas_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Stat_Seas_KW_try$Depth_C <- factor(even_Stat_Seas_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_evenness_Seas_Stat_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Seas_Stat_boxplot <- ggplot(even_Stat_Seas_KW_try, aes(x = Stat_Seas, y = mean))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = even_Stat_Seas_KW_sigs, aes(label = siglabel, x = Stat_Seas, y = 0), size = 2) +
  scale_color_manual(name = "season", values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),
                     labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                     labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
    scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa"),
                   labels=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa")) + 
  xlab("") + ylab("Simpson' Evenness") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Seas_Stat_boxplot #top, right, bottom, left
dev.off()

## 2. evenness differences based on Fraction and Season
even_Frac_Seas_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Frac_Seas)) 
print(even_Frac_Seas_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Frac_Seas_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Frac_Seas))  ## Defaults to P < 0.05
print(even_Frac_Seas_KW_MC)

# Test to find the letters to represent significance in a plot
even_Frac_Seas_KW_test <- even_Frac_Seas_KW_MC$dif.com$difference # select logical vector
names(even_Frac_Seas_KW_test) <- row.names(even_Frac_Seas_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Frac_Seas_KW_letters <- multcompLetters(even_Frac_Seas_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Frac_Seas_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Frac_Seas_KW_letters$Letters)), as.vector(even_Frac_Seas_KW_letters$Letters))
colnames(even_Frac_Seas_KW_sigs_dataframe) <- c("Frac_Seas", "siglabel")
even_Frac_Seas_KW_try <- merge(simps_even, even_Frac_Seas_KW_sigs_dataframe, by = "Frac_Seas")

even_Frac_Seas_KW_try$Frac_Seas <-factor(even_Frac_Seas_KW_try$Frac_Seas,levels=c( "B Sp", "B Su", "B Fa", "E Sp", "E Su", "E Fa"))

even_Frac_Seas_KW_sigs <- even_Frac_Seas_KW_try %>%
  select(Frac_Seas, Frac_Seas, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Frac_Seas_KW_try$Fraction <- factor(even_Frac_Seas_KW_try$Fraction,levels=c("B","E"))
even_Frac_Seas_KW_try$Season <- factor(even_Frac_Seas_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Frac_Seas_KW_try$Station <- factor(even_Frac_Seas_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Frac_Seas_KW_try$Depth_C <- factor(even_Frac_Seas_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_evenness_Frac_Seas_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Frac_Seas_boxplot <- ggplot(even_Frac_Seas_KW_try, aes(x = Frac_Seas, y = mean))  +
  geom_boxplot(aes(color=Season),outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = even_Frac_Seas_KW_sigs, aes(label = siglabel, x = Frac_Seas, y = 0), size = 2) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                      labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
 scale_x_discrete(breaks=c("B Sp", "B Su", "B Fa", "E Sp", "E Su", "E Fa"),
                   labels=c("FL Sp", "FL Su", "FL Fa", "PA Sp", "PA Su", "PA Fa")) + 
   xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Frac_Seas_boxplot #top, right, bottom, left
dev.off()


## 3. evenness differences based on Station and Season
#  Resort the factors of Season & Fraction object to have primary sorting by station
even_Stat_Seas_KW_try$Stat_Seas <-factor(even_Stat_Seas_KW_try$Stat_Seas,levels=c( "MLB Sp","MLB Su","MLB Fa","MM15 Sp","MM15 Su","MM15 Fa","MM110 Sp","MM110 Su","MM110 Fa"))

even_Stat_Seas_KW_sigs <- even_Stat_Seas_KW_try %>%
  select(Stat_Seas, Stat_Seas, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Station_KW_try$Fraction <- factor(even_Station_KW_try$Fraction,levels=c("B","E"))
even_Station_KW_try$Season <- factor(even_Station_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Station_KW_try$Station <- factor(even_Station_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Station_KW_try$Depth_C <- factor(even_Station_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_evenness_Stat_Seas_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Stat_Seas_boxplot <- ggplot(even_Stat_Seas_KW_try, aes(x = Stat_Seas, y = mean))  +
  geom_boxplot(aes(color=Season),outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = even_Stat_Seas_KW_sigs, aes(label = siglabel, x = Stat_Seas, y = 0), size = 2) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                      labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa"),
                   labels=c("MLB Sp", "MM15 Sp", "MM110 Sp","MLB Su", "MM15 Su", "MM110 Su","MLB Fa", "MM15 Fa", "MM110 Fa")) +   xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Stat_Seas_boxplot #top, right, bottom, left
dev.off()


## 4. evenness differences based on Depth_C and Season
#simps_even_off <- simps_even[simps_even$Station == "MM110" & simps_even$Season == "Su", ]

even_Depth_Seas_KW <- kruskal.test(simps_even$mean ~ as.factor(simps_even$Depth_Seas)) 
print(even_Depth_Seas_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
even_Depth_Seas_KW_MC <- kruskalmc(simps_even$mean ~ as.factor(simps_even$Depth_Seas))  ## Defaults to P < 0.05
print(even_Depth_Seas_KW_MC)

# Test to find the letters to represent significance in a plot
even_Depth_Seas_KW_test <- even_Depth_Seas_KW_MC$dif.com$difference # select logical vector
names(even_Depth_Seas_KW_test) <- row.names(even_Depth_Seas_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
even_Depth_Seas_KW_letters <- multcompLetters(even_Depth_Seas_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
even_Depth_Seas_KW_sigs_dataframe <-  data.frame(as.vector(names(even_Depth_Seas_KW_letters$Letters)), as.vector(even_Depth_Seas_KW_letters$Letters))
colnames(even_Depth_Seas_KW_sigs_dataframe) <- c("Depth_Seas", "siglabel")
even_Depth_Seas_KW_try <- merge(simps_even, even_Depth_Seas_KW_sigs_dataframe, by = "Depth_Seas")

even_Depth_Seas_KW_try$Depth_Seas <-factor(even_Depth_Seas_KW_try$Depth_Seas,levels=c( "S Sp", "S Su", "S Fa", "M Su", "D Sp", "D Su", "D Fa"))

even_Depth_Seas_KW_sigs <- even_Depth_Seas_KW_try %>%
  select(Depth_Seas, Depth_Seas, siglabel) %>%
  distinct()

# finally, the plot
alpha_title <- paste("evenness when rarefied at",min(sample_sums(physeq_dupmerge)),"reads")
even_Depth_Seas_KW_try$Fraction <- factor(even_Depth_Seas_KW_try$Fraction,levels=c("B","E"))
even_Depth_Seas_KW_try$Season <- factor(even_Depth_Seas_KW_try$Season,levels=c("Sp","Su","Fa"))
even_Depth_Seas_KW_try$Station <- factor(even_Depth_Seas_KW_try$Station,levels=c("MLB","MM15","MM110"))
even_Depth_Seas_KW_try$Depth_C <- factor(even_Depth_Seas_KW_try$Depth_C,levels=c("S","M","D"))

pdf(file="Figures/Rplot_evenness_Depth_Seas_boxplot.pdf", width= 5, height=5, pointsize= 8)
even_Depth_Seas_boxplot <- ggplot(even_Depth_Seas_KW_try, aes(x = Depth_Seas, y = mean))  +
  geom_boxplot(aes(color=Season),outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=Season,shape = Station,size = Fraction,fill=Depth_Seas), position = position_dodge(0.8)) +
  geom_text(data = even_Depth_Seas_KW_sigs, aes(label = siglabel, x = Depth_Seas, y = 0), size = 2) +
  scale_color_manual(name = "season", breaks = c("Sp","Su","Fa"),values = c("Sp" = "darkgreen", "Su" = "red", "Fa" = "mediumblue"),labels = c("Spring",  "Summer", "Fall")) +
  scale_shape_manual(name = "station", values = c("MLB" = 21, "MM15" = 22, "MM110" = 24),
                      labels = c("Estuary", "Near",  "Off")) + 
  scale_size_manual(name = "fraction", breaks = c("B","E"), 
                     labels = c("Free","Particle"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  scale_x_discrete(breaks=c("S Sp", "S Su", "S Fa", "M Su", "D Sp", "D Su", "D Fa"),
                   labels=c("S Sp", "S Su", "S Fa", "M Su", "D Sp", "D Su", "D Fa")) + 
  xlab("") + ylab("") + theme_bw() + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="none", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); even_Depth_Seas_boxplot #top, right, bottom, left
dev.off()
```

```{r final figure 2}
library(grid)
library(gridExtra)

pdf(file="Figures/Rplot_AlphaDiv_boxplot.pdf", width= 7, height=5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(rich_Season_boxplot,rich_Station_boxplot,rich_Fraction_boxplot,rich_Depth_C_boxplot,even_Season_boxplot,even_Station_boxplot,even_Fraction_boxplot,even_Depth_C_boxplot,ncol=4)
dev.off()

pdf(file="Figures/Rplot_AlphaDiv_multifact_boxplot.pdf", width= 7, height=5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(rich_Seas_Stat_boxplot,rich_Stat_Seas_boxplot,rich_Frac_Seas_boxplot,rich_Depth_Seas_boxplot,even_Seas_Stat_boxplot,even_Stat_Seas_boxplot,even_Frac_Seas_boxplot,even_Depth_Seas_boxplot,ncol=4)
dev.off()
```

### Figure 3 + Table 2: PCoA, PERMANOVA
```{r adonis}
#calculate average number of reads in merged, not scaled data (as this is what we will use to prune taxa in deseq and we want to have ordinations based on same taxa)
mean(sample_sums(physeq_dupmerge))

# Keep OTUs above average relative abundance of 330 reads (0.1% before scaling) for use with deseq 
a <- apply(otu_table(physeq_dupmerge), MARGIN = 1, function(x) {mean(x)} )
keep <- a > 330
physeq_dupmerge.prune <- prune_taxa(keep, physeq_dupmerge)
ntaxa(physeq_dupmerge.prune) 

# Keep OTUs above average relative abundance of 0.1% in scaled physeq object
a <- apply(otu_table(physeq.scale), MARGIN = 1, function(x) {mean(x)} )
keep <- row.names(otu_table(physeq_dupmerge.prune))
physeq.prune <- prune_taxa(keep, physeq.scale)
ntaxa(physeq.prune) 

# Calculate bray curtis
physeq.bray <- phyloseq::distance(physeq = physeq.prune, method = "bray")

### Let's run ADONIS using Michelle Berry's MicrobeMiseq function --- FOR TABLE 2
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray","Fraction")
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray","Season")
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray","Station")
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray","Depth_C")
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray","DayNight")
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray","Lake")
formula<-c("Fraction","Season","Lake","Station","Depth_C","DayNight")
phyloseq_to_adonis(physeq.scale, distmat = physeq.bray, dist = "bray",formula)

# Calculate Sorensen
physeq.sor <- phyloseq::distance(physeq = physeq.prune, method = "bray", binary=TRUE)

### Let's run ADONIS using Michelle Berry's MicrobeMiseq function --- FOR TABLE 2
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen","Fraction")
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen","Season")
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen","Station")
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen","Depth_C")
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen","DayNight")
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen","Lake")
formula<-c("Fraction","Season","Lake","Station","Depth_C","DayNight")
phyloseq_to_adonis(physeq.scale, distmat = physeq.sor, dist = "sorensen",formula)
```

```{r pcoa bray-curtis}
# Betadiv pcoa Bray-Curtis DNA only 
physeq.pcoa <-
  ordinate(
    physeq = physeq.scale,
    method = "PCoA",
    distance = "bray"
  )

physeq.pcoa.vectors <- data.frame(physeq.pcoa$vectors[, 1:4])
physeq.pcoa.vectors$Duplicates <- row.names(physeq.pcoa.vectors)
physeq.pcoa.df <- merge(physeq.pcoa.vectors,data.frame(duplicate_metadata),by="Duplicates")
for(i in 1:length(physeq.pcoa.df$Season)){
  physeq.pcoa.df$Depth_Seas[i]<-paste(as.character(physeq.pcoa.df$Depth_C[i]),
                                  as.character(physeq.pcoa.df$Season[i]))}
bray_values_DNA <- physeq.pcoa$values
bray_rel_eigens_DNA <- bray_values_DNA$Relative_eig
bray_rel_eigen1_DNA <- bray_rel_eigens_DNA[1]
bray_rel_eigen1_percent_DNA <- round(bray_rel_eigen1_DNA * 100, digits = 1)
bray_rel_eigen2_DNA <- bray_rel_eigens_DNA[2]
bray_rel_eigen2_percent_DNA <- round(bray_rel_eigen2_DNA * 100, digits = 1)
bray_rel_eigen3_DNA <- bray_rel_eigens_DNA[3]
bray_rel_eigen3_percent_DNA <- round(bray_rel_eigen3_DNA * 100, digits = 1)
bray_rel_eigen4_DNA <- bray_rel_eigens_DNA[4]
bray_rel_eigen4_percent_DNA <- round(bray_rel_eigen4_DNA * 100, digits = 1)
bray_axis1_DNA <- paste("PCoA1:",bray_rel_eigen1_percent_DNA,"%")
bray_axis2_DNA <- paste("PCoA2:",bray_rel_eigen2_percent_DNA,"%")
bray_axis3_DNA <- paste("PCoA3:",bray_rel_eigen3_percent_DNA,"%")
bray_axis4_DNA <- paste("PCoA4:",bray_rel_eigen4_percent_DNA,"%")

PCoA_DNA_title <- paste("Bray-Curtis,",ntaxa(physeq.scale),"OTUs")

pdf(file="Figures/Rplot_PCOA12_BC_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_BC12_DNA <- ggplot(physeq.pcoa.df, aes(Axis.1, Axis.2, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(bray_axis1_DNA) + ylab(bray_axis2_DNA) + ggtitle(PCoA_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_BC12_DNA
  dev.off()

pdf(file="Figures/Rplot_PCOA13_BC_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_BC13_DNA <- ggplot(physeq.pcoa.df, aes(Axis.1, Axis.3, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(bray_axis1_DNA) + ylab(bray_axis3_DNA) + ggtitle(PCoA_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_BC13_DNA
  dev.off()

```

```{r pcoa bray curtis after removal otu <0.1% on average}
# Prune out rare taxa
#calculate average number of reads in merged, not scaled data (as this is what we will use to prune taxa in deseq and we want to have ordinations based on same taxa)
mean(sample_sums(physeq_dupmerge))

# Keep OTUs above average relative abundance of 330 reads (0.1% before scaling) for use with deseq 
a <- apply(otu_table(physeq_dupmerge), MARGIN = 1, function(x) {mean(x)} )
keep <- a > 330
physeq_dupmerge.prune <- prune_taxa(keep, physeq_dupmerge)
ntaxa(physeq_dupmerge.prune) 

# Keep OTUs above average relative abundance of 0.1% in scaled physeq object
a <- apply(otu_table(physeq.scale), MARGIN = 1, function(x) {mean(x)} )
keep <- row.names(otu_table(physeq_dupmerge.prune))
physeq.prune <- prune_taxa(keep, physeq.scale)
ntaxa(physeq.prune) 

### determine how many sequences included for each sample when only considering top OTUs
relab_toptaxa <- sample_sums(physeq.prune)/sample_sums(physeq.scale)*100
mean(relab_toptaxa)
sd(relab_toptaxa)

sample_sums(physeq)

# Betadiv pcoa Bray-Curtis DNA only 
physeq.pcoa.prune <-
  ordinate(
    physeq = physeq.prune,
    method = "PCoA",
    distance = "bray"
  )

physeq.pcoa.prune.vectors <- data.frame(physeq.pcoa.prune$vectors[, 1:4])
physeq.pcoa.prune.vectors$Duplicates <- row.names(physeq.pcoa.prune.vectors)
physeq.pcoa.prune.df <- merge(physeq.pcoa.prune.vectors,data.frame(duplicate_metadata),by="Duplicates")
for(i in 1:length(physeq.pcoa.prune.df$Season)){
  physeq.pcoa.prune.df$Depth_Seas[i]<-paste(as.character(physeq.pcoa.prune.df$Depth_C[i]),
                                  as.character(physeq.pcoa.prune.df$Season[i]))}
for(i in 1:length(physeq.pcoa.prune.df$Season)){
  physeq.pcoa.prune.df$Frac_Seas[i]<-paste(as.character(physeq.pcoa.prune.df$Fraction[i]),
                                  as.character(physeq.pcoa.prune.df$Season[i]))}
for(i in 1:length(physeq.pcoa.prune.df$Station)){
  physeq.pcoa.prune.df$Depth_Stat[i]<-paste(as.character(physeq.pcoa.prune.df$Depth_C[i]),
                                  as.character(physeq.pcoa.prune.df$Station[i]))}

bray_values_DNA <- physeq.pcoa.prune$values
bray_rel_eigens_DNA <- bray_values_DNA$Relative_eig
bray_rel_eigen1_DNA <- bray_rel_eigens_DNA[1]
bray_rel_eigen1_percent_DNA <- round(bray_rel_eigen1_DNA * 100, digits = 1)
bray_rel_eigen2_DNA <- bray_rel_eigens_DNA[2]
bray_rel_eigen2_percent_DNA <- round(bray_rel_eigen2_DNA * 100, digits = 1)
bray_rel_eigen3_DNA <- bray_rel_eigens_DNA[3]
bray_rel_eigen3_percent_DNA <- round(bray_rel_eigen3_DNA * 100, digits = 1)
bray_rel_eigen4_DNA <- bray_rel_eigens_DNA[4]
bray_rel_eigen4_percent_DNA <- round(bray_rel_eigen4_DNA * 100, digits = 1)
bray_axis1_DNA <- paste("PCoA1:",bray_rel_eigen1_percent_DNA,"%")
bray_axis2_DNA <- paste("PCoA2:",bray_rel_eigen2_percent_DNA,"%")
bray_axis3_DNA <- paste("PCoA3:",bray_rel_eigen3_percent_DNA,"%")
bray_axis4_DNA <- paste("PCoA4:",bray_rel_eigen4_percent_DNA,"%")

PCoA_DNA_title <- paste("Bray-Curtis,",ntaxa(physeq.prune),"OTUs")

pdf(file="Figures/Rplot_PCOA12_BC.PRUNE_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_BC12.PRUNE_DNA <- ggplot(physeq.pcoa.prune.df, aes(Axis.1, Axis.2, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(bray_axis1_DNA) + ylab(bray_axis2_DNA) + ggtitle(PCoA_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_BC12.PRUNE_DNA
  dev.off()

pdf(file="Figures/Rplot_PCOA13_BC_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_BC13.PRUNE_DNA <- ggplot(physeq.pcoa.prune.df, aes(Axis.1, Axis.3, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(bray_axis1_DNA) + ylab(bray_axis3_DNA) + ggtitle(PCoA_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_BC13.PRUNE_DNA
  dev.off()

pcoa_BC12.PRUNE_DNA_PAFLSeas <- ggplot(physeq.pcoa.prune.df, aes(Axis.1, Axis.2, color = Season, fill = Frac_Seas)) +
  xlab(bray_axis1_DNA) + ylab(bray_axis2_DNA) + ggtitle(PCoA_DNA_title) +
  geom_point(alpha=0.9, shape=21) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_fill_manual(name = "depth", breaks = c("B Sp", "B Su", "B Fa", "E Sp", "E Su", "E Fa"), 
                     labels = c("FL","FL","FL", "PA", "PA", "PA"),
                     values = c("B Fa" = "mediumblue", "B Sp" = "darkgreen", "B Su" = "red", "E Sp" = "white", "E Su" = "white", "E Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_BC12.PRUNE_DNA_PAFLSeas

pcoa_BC12.PRUNE_DNA_DepthStat <- ggplot(physeq.pcoa.prune.df, aes(Axis.1, Axis.2, color = Station, fill = Depth_Stat)) +
  xlab(bray_axis1_DNA) + ylab(bray_axis2_DNA) + ggtitle(PCoA_DNA_title) +
  geom_point(alpha=0.9, shape=21) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = "darkolivegreen4", "MM15" = "darkorange", "MM110" = "deepskyblue4")) +
  scale_fill_manual(name = "depth", breaks = c("D MLB", "D MM15", "D MM110", "M MM110", "S MLB", "S MM15", "S MM110"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D MLB" = "white", "D MM15" = "white", "D MM110" = "white", "M MM110" = "deepskyblue2", "S MLB" = "darkolivegreen4", "S MM15" = "darkorange", "S MM110" = "deepskyblue4")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_BC12.PRUNE_DNA_DepthStat
```

```{r pcoa SÃ¸rensen}
# Betadiv pcoa SÃ¸rensen DNA only 
physeq.pcoa.sor <-
  ordinate(
    physeq = physeq.scale,
    method = "PCoA",
    distance = "bray",
    binary = TRUE
  )

physeq.pcoa.sor.vectors <- data.frame(physeq.pcoa.sor$vectors[, 1:4])
physeq.pcoa.sor.vectors$Duplicates <- row.names(physeq.pcoa.sor.vectors)
physeq.pcoa.sor.df <- merge(physeq.pcoa.sor.vectors,data.frame(duplicate_metadata),by="Duplicates")
for(i in 1:length(physeq.pcoa.sor.df$Season)){
  physeq.pcoa.sor.df$Depth_Seas[i]<-paste(as.character(physeq.pcoa.sor.df$Depth_C[i]),
                                  as.character(physeq.pcoa.sor.df$Season[i]))}
sor_values_DNA <- physeq.pcoa.sor$values
sor_rel_eigens_DNA <- sor_values_DNA$Relative_eig
sor_rel_eigen1_DNA <- sor_rel_eigens_DNA[1]
sor_rel_eigen1_percent_DNA <- round(sor_rel_eigen1_DNA * 100, digits = 1)
sor_rel_eigen2_DNA <- sor_rel_eigens_DNA[2]
sor_rel_eigen2_percent_DNA <- round(sor_rel_eigen2_DNA * 100, digits = 1)
sor_rel_eigen3_DNA <- sor_rel_eigens_DNA[3]
sor_rel_eigen3_percent_DNA <- round(sor_rel_eigen3_DNA * 100, digits = 1)
sor_rel_eigen4_DNA <- sor_rel_eigens_DNA[4]
sor_rel_eigen4_percent_DNA <- round(sor_rel_eigen4_DNA * 100, digits = 1)
sor_axis1_DNA <- paste("PCoA1:",sor_rel_eigen1_percent_DNA,"%")
sor_axis2_DNA <- paste("PCoA2:",sor_rel_eigen2_percent_DNA,"%")
sor_axis3_DNA <- paste("PCoA3:",sor_rel_eigen3_percent_DNA,"%")
sor_axis4_DNA <- paste("PCoA4:",sor_rel_eigen4_percent_DNA,"%")

PCoA_Sor_DNA_title <- paste("SÃ¸rensen,",ntaxa(physeq.scale),"OTUs")

pdf(file="Figures/Rplot_PCOA12_SOR_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_SOR12_DNA <- ggplot(physeq.pcoa.sor.df, aes(Axis.1, -Axis.2, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(sor_axis1_DNA) + ylab(sor_axis2_DNA) + ggtitle(PCoA_Sor_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_SOR12_DNA
  dev.off()

pdf(file="Figures/Rplot_PCOA13_SOR_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_SOR13_DNA <- ggplot(physeq.pcoa.sor.df, aes(Axis.1, Axis.3, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(sor_axis1_DNA) + ylab(sor_axis3_DNA) + ggtitle(PCoA_Sor_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_SOR13_DNA
  dev.off()

```

```{r pcoa SÃ¸rensen after removal of OTU < 0.01% on average}
#calculate average number of reads in merged, not scaled data (as this is what we will use to prune taxa in deseq and we want to have ordinations based on same taxa)
mean(sample_sums(physeq_dupmerge))

# Keep OTUs above average relative abundance of 330 reads (0.1% before scaling) for use with deseq 
a <- apply(otu_table(physeq_dupmerge), MARGIN = 1, function(x) {mean(x)} )
keep <- a > 330
physeq_dupmerge.prune <- prune_taxa(keep, physeq_dupmerge)
ntaxa(physeq_dupmerge.prune) 

# Keep OTUs above average relative abundance of 0.1% in scaled physeq object
a <- apply(otu_table(physeq.scale), MARGIN = 1, function(x) {mean(x)} )
keep <- row.names(otu_table(physeq_dupmerge.prune))
physeq.prune <- prune_taxa(keep, physeq.scale)
ntaxa(physeq.prune) 

# Betadiv pcoa SÃ¸rensen DNA only 
physeq.pcoa.prune.sor <-
  ordinate(
    physeq = physeq.prune,
    method = "PCoA",
    distance = "bray",
    binary = TRUE
  )

physeq.pcoa.prune.sor.vectors <- data.frame(physeq.pcoa.prune.sor$vectors[, 1:4])
physeq.pcoa.prune.sor.vectors$Duplicates <- row.names(physeq.pcoa.prune.sor.vectors)
physeq.pcoa.prune.sor.df <- merge(physeq.pcoa.prune.sor.vectors,data.frame(duplicate_metadata),by="Duplicates")
for(i in 1:length(physeq.pcoa.prune.sor.df$Season)){
  physeq.pcoa.prune.sor.df$Depth_Seas[i]<-paste(as.character(physeq.pcoa.prune.sor.df$Depth_C[i]),
                                  as.character(physeq.pcoa.prune.sor.df$Season[i]))}
sor_values_DNA <- physeq.pcoa.prune.sor$values
sor_rel_eigens_DNA <- sor_values_DNA$Relative_eig
sor_rel_eigen1_DNA <- sor_rel_eigens_DNA[1]
sor_rel_eigen1_percent_DNA <- round(sor_rel_eigen1_DNA * 100, digits = 1)
sor_rel_eigen2_DNA <- sor_rel_eigens_DNA[2]
sor_rel_eigen2_percent_DNA <- round(sor_rel_eigen2_DNA * 100, digits = 1)
sor_rel_eigen3_DNA <- sor_rel_eigens_DNA[3]
sor_rel_eigen3_percent_DNA <- round(sor_rel_eigen3_DNA * 100, digits = 1)
sor_rel_eigen4_DNA <- sor_rel_eigens_DNA[4]
sor_rel_eigen4_percent_DNA <- round(sor_rel_eigen4_DNA * 100, digits = 1)
sor_axis1_DNA <- paste("PCoA1:",sor_rel_eigen1_percent_DNA,"%")
sor_axis2_DNA <- paste("PCoA2:",sor_rel_eigen2_percent_DNA,"%")
sor_axis3_DNA <- paste("PCoA3:",sor_rel_eigen3_percent_DNA,"%")
sor_axis4_DNA <- paste("PCoA4:",sor_rel_eigen4_percent_DNA,"%")

PCoA_Sor_DNA_title <- paste("SÃ¸rensen,",ntaxa(physeq.prune),"OTUs")

pdf(file="Figures/Rplot_PCOA12_SOR.PRUNE_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_SOR12.PRUNE_DNA <- ggplot(physeq.pcoa.prune.sor.df, aes(-Axis.1, -Axis.2, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(sor_axis1_DNA) + ylab(sor_axis2_DNA) + ggtitle(PCoA_Sor_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_SOR12.PRUNE_DNA
  dev.off()

pdf(file="Figures/Rplot_PCOA13_SOR.PRUNE_DNA.pdf", width= 20, height=15, pointsize= 8)
pcoa_SOR13.PRUNE_DNA <- ggplot(physeq.pcoa.prune.sor.df, aes(-Axis.1, Axis.3, color = Season, shape = Station, size = Fraction, fill = Depth_Seas)) +
  xlab(sor_axis1_DNA) + ylab(sor_axis3_DNA) + ggtitle(PCoA_Sor_DNA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "season", breaks=c("Sp", "Su",  "Fa"),
                     labels = c("Spring", "Summer",  "Fall"), 
                     values = c("Fa" = "mediumblue", "Sp" = "darkgreen", "Su" = "red")) +
  scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
  scale_size_manual(name = "fraction", breaks = c("E", "B"), 
                     labels = c("Particle", "Free"),
                     values = c("B" = 1, "E" = 2)) +
  scale_fill_manual(name = "depth", breaks = c("D Sp", "D Su", "D Fa", "M Su", "S Sp", "S Su", "S Fa"), 
                     labels = c("Deep","Deep","Deep", "Chla max", "Surface", "Surface", "Surface"),
                     values = c("D Fa" = "mediumblue", "D Sp" = "darkgreen", "D Su" = "red", "M Su" = "orange", "S Sp" = "white", "S Su" = "white", "S Fa" = "white")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_SOR13.PRUNE_DNA
  dev.off()

```

```{r final figures 3 + S2}
library(grid)
library(gridExtra)

pdf(file="Figures/Rplot_BetaDiv_0.1_pcoa_MainFig3.pdf", width= 7, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(pcoa_BC12.PRUNE_DNA_PAFLSeas,pcoa_BC12.PRUNE_DNA_DepthStat,ncol=2)
dev.off()

pdf(file="Figures/Rplot_BetaDiv_0.1vsall_pcoa.pdf", width= 7, height=7, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(pcoa_BC12.PRUNE_DNA,pcoa_SOR12.PRUNE_DNA,pcoa_BC12_DNA,pcoa_SOR12_DNA,ncol=4)
dev.off()
```

## Figure 4: boxplots bray-curtis distances across seasons and stations by fraction 
```{r create dataframe with bray curtis dissimilarities and factors describing comparison classes}

#create dataframe with paired BC dissimilarity and add metadata for compared samples
physeq.bray <- phyloseq::distance(physeq = physeq.prune, method = "bray")
bray_df <- melt(as.matrix(physeq.bray), varnames = c("samp1", "samp2"))

bray_df$season1 <- substr(bray_df$samp1,1,2) # Create a new column called season1 with first 2 letters of string
bray_df$season2 <- substr(bray_df$samp2,1,2) # Create a new column called season2 with first 2 letters of string
bray_df$fraction1 <- substr(bray_df$samp1,6,6) # Create a new column called fraction1 with letter 6 of string
bray_df$fraction2 <- substr(bray_df$samp2,6,6) # Create a new column called fraction2 with letter 6 of string
bray_df$station1 <- unlist(strsplit(toString(bray_df$samp1), "[., ]"))[seq(3,length(unlist(strsplit(toString(bray_df$samp1), split="[., ]"))),6)] # Create a new column called station1 with staton name
bray_df$station2 <- unlist(strsplit(toString(bray_df$samp2), "[., ]"))[seq(3,length(unlist(strsplit(toString(bray_df$samp2), split="[., ]"))),6)] # Create a new column called station2 with station name
bray_df$depth1 <- substr(unlist(strsplit(toString(bray_df$samp1), "[., ]"))[seq(4,length(unlist(strsplit(toString(bray_df$samp1), split="[., ]"))),6)],1,1) # Create a new column called depth1 with depth
bray_df$depth2 <- substr(unlist(strsplit(toString(bray_df$samp2), "[., ]"))[seq(4,length(unlist(strsplit(toString(bray_df$samp2), split="[., ]"))),6)],1,1) # Create a new column called depth2 with depth

#combine category names to group similar comparisons - sorry, clunky code here, but hey, it works!
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp_ML"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp_LM"

bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp_MLB"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp_MM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp_MM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Sp_FL"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Sp_PA"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Sp_FLvsPA"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Sp_FLvsPA"

bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Sp_Surf"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Sp_Deep"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Sp_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Sp"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Sp_SurfvsDeep"

bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Su_ML"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Su_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Su_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Su_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Su_LM"

bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Su_MLB"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Su_MM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Su_MM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Su_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Su_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Su_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Su_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Su_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Su_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Su_FL"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Su_PA"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Su_FLvsPA"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Su_FLvsPA"

bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Su_Surf"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Su_Deep"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Su_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Su"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Su_SurfvsDeep"

bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Fa_ML"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Fa_LM"

bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Fa_MLB"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Fa_MM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Fa_MM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Fa_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Fa_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Fa_FL"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Fa_PA"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Fa_FLvsPA"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Fa_FLvsPA"

bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Fa_Surf"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Fa_Deep"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Fa_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Fa"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Fa_SurfvsDeep"

bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Su_MLB"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Su_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Su_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Su_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Su_LM"

bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Su_MLB"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Su_MM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Su_MM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Su_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Su_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Su_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Su_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Sp-Su_FL"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Sp-Su_PA"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Sp-Su_FLvsPA"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Sp-Su_FLvsPA"

bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Sp-Su_Surf"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Sp-Su_Deep"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Sp-Su_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Su"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Sp-Su_SurfvsDeep"

bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Su_MLB"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Su_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Su_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Su_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Su_LM"

bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Su_MLB"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Su_MM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Su_MM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Su_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Su_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Su_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Su_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Su_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Sp-Su_FL"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Sp-Su_PA"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Sp-Su_FLvsPA"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Sp-Su_FLvsPA"

bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Sp-Su_Surf"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Sp-Su_Deep"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Sp-Su_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Sp"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Sp-Su_SurfvsDeep"

bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Fa_MLB"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Fa_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Fa_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Fa_LM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Fa_LM"

bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Fa_MLB"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Fa_MM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Fa_MM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Fa_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Fa_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Sp-Fa_FL"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Sp-Fa_PA"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Sp-Fa_FLvsPA"
bray_df$compfrac[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Sp-Fa_FLvsPA"

bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Sp-Fa_Surf"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Sp-Fa_Deep"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Sp-Fa_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Sp"&bray_df$season2=="Fa"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Sp-Fa_SurfvsDeep"

bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Fa_MLB"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Fa_LM"

bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Sp-Fa_MLB"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Sp-Fa_MM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Sp-Fa_MM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Sp-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Sp-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Sp-Fa_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Sp-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Sp-Fa_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Sp-Fa_FL"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Sp-Fa_PA"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Sp-Fa_FLvsPA"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Sp-Fa_FLvsPA"

bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Sp-Fa_Surf"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Sp-Fa_Deep"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Sp-Fa_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Sp"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Sp-Fa_SurfvsDeep"

bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Su-Fa_MLB"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Su-Fa_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Su-Fa_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Su-Fa_LM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Su-Fa_LM"

bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Su-Fa_MLB"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Su-Fa_MM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Su-Fa_MM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Su-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Su-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Su-Fa_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Su-Fa_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Su-Fa_FL"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Su-Fa_PA"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Su-Fa_FLvsPA"
bray_df$compfrac[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Su-Fa_FLvsPA"

bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Su-Fa_Surf"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Su-Fa_Deep"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Su-Fa_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Su"&bray_df$season2=="Fa"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Su-Fa_SurfvsDeep"

bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Su-Fa_MLB"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Su-Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Su-Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Su-Fa_LM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsLM"
bray_df$complake[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Su-Fa_LM"

bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MLB"]<-"Su-Fa_MLB"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM15"]<-"Su-Fa_MM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM110"]<-"Su-Fa_MM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM15"]<-"Su-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MLB"&bray_df$station2=="MM110"]<-"Su-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MM110"]<-"Su-Fa_MM15vsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM15"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsMM15"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MLB"]<-"Su-Fa_MLBvsMM110"
bray_df$compstat[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$station1=="MM110"&bray_df$station2=="MM15"]<-"Su-Fa_MM15vsMM110"

bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$fraction1=="B"&bray_df$fraction2=="B"]<-"Su-Fa_FL"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$fraction1=="E"&bray_df$fraction2=="E"]<-"Su-Fa_PA"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$fraction1=="B"&bray_df$fraction2=="E"]<-"Su-Fa_FLvsPA"
bray_df$compfrac[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$fraction1=="E"&bray_df$fraction2=="B"]<-"Su-Fa_FLvsPA"

bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$depth1=="S"&bray_df$depth2=="S"]<-"Su-Fa_Surf"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$depth1=="D"&bray_df$depth2=="D"]<-"Su-Fa_Deep"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$depth1=="S"&bray_df$depth2=="D"]<-"Su-Fa_SurfvsDeep"
bray_df$compdepth[bray_df$season1=="Fa"&bray_df$season2=="Su"&bray_df$depth1=="D"&bray_df$depth2=="S"]<-"Su-Fa_SurfvsDeep"


#removing self comparisons and calculating average per group
bray_df <- subset(bray_df, samp1 != samp2)
```

```{r plot FL vs PA fraction}
#only retain BC dissimilarities within each fraction, not betwee the two fractions
bray_df_wifrac <- subset(bray_df, fraction1 == fraction2)

#Kruskall-Wallis test
bray_frac_KW <- kruskal.test(bray_df_wifrac$value ~ as.factor(bray_df_wifrac$fraction1)) 
print(bray_frac_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
bray_frac_KW_MC <- kruskalmc(bray_df_wifrac$value ~ as.factor(bray_df_wifrac$fraction1))  ## Defaults to P < 0.05
print(bray_frac_KW_MC)

# Test to find the letters to represent significance in a plot
bray_frac_KW_test <- bray_frac_KW_MC$dif.com$difference # select logical vector
names(bray_frac_KW_test) <- row.names(bray_frac_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
bray_frac_KW_letters <- multcompLetters(bray_frac_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
bray_frac_KW_sigs_dataframe <-  data.frame(as.vector(names(bray_frac_KW_letters$Letters)), as.vector(bray_frac_KW_letters$Letters))
colnames(bray_frac_KW_sigs_dataframe) <- c("fraction1", "siglabel")
bray_frac_KW_try <- merge(bray_df_wifrac, bray_frac_KW_sigs_dataframe, by = "fraction1")

bray_frac_KW_try$fraction1 <-factor(bray_frac_KW_try$fraction1,levels=c( "B", "E"))

bray_frac_KW_sigs <- bray_frac_KW_try %>%
  select(fraction1, fraction1, siglabel) %>%
  distinct()

#set order of comparions to display
#bray_df_wifrac$complake <- factor(bray_df_wifrac$complake,levels=c("Sp_LM","Sp_MLBvsLM","Su_ML","Su_LM","Su_MLBvsLM","Fa_ML","Fa_LM","Fa_MLBvsLM","Sp-Su_MLB","Sp-Fa_MLB","Su-Fa_MLB","Sp-Su_LM","Sp-Fa_LM","Su-Fa_LM","Sp-Su_MLBvsLM","Sp-Fa_MLBvsLM","Su-Fa_MLBvsLM"))

bray_df_wifrac$complake <- factor(bray_df_wifrac$complake,levels=c("Su_ML","Fa_ML","Sp_LM","Su_LM","Fa_LM","Sp_MLBvsLM","Su_MLBvsLM","Fa_MLBvsLM","Sp-Su_MLB","Sp-Fa_MLB","Su-Fa_MLB","Sp-Su_LM","Sp-Fa_LM","Su-Fa_LM","Sp-Su_MLBvsLM","Sp-Fa_MLBvsLM","Su-Fa_MLBvsLM"))

#determine number of samples in FL and PA comparisons
nrow(subset(bray_df_wifrac, fraction1 == "B"))
nrow(subset(bray_df_wifrac, fraction1 == "E"))

#plot
pdf(file="Figures/Rplot_BetaDivFLvsPA_boxplot.pdf", width= 3.5, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
bray_wifrac_boxplot <- ggplot(bray_df_wifrac, aes(x = fraction1, y = value))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=complake),size=1,alpha=1, position = position_dodge(0.8)) +
  geom_text(data = bray_frac_KW_sigs, aes(label = siglabel, x = fraction1, y = 0), size = 2) +
  xlab("") + ylab("Bray-Curtis dissimilarity") + theme_bw() + ylim(0,1) +
  scale_x_discrete(breaks=c("B", "E"),labels=c("FL\n(n=552)","PA\n(n=506)")) + 
  guides(color=guide_legend(title="comparison")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 6, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 4),  #Set the legend text
                                 legend.position="right", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); bray_wifrac_boxplot #top, right, bottom, left
dev.off()
```

```{r plot station comparisons FL fraction}
#only retain BC dissimilarities within each fraction, not betwee the two fractions
bray_df_wistat <- subset(bray_df, station1 == station2)
bray_df_wistat <- subset(bray_df_wistat, fraction1 == fraction2)
bray_df_wistat <- subset(bray_df_wistat, season1 != season2)
bray_df_wistat <- subset(bray_df_wistat, fraction1 == "B")

#Kruskall-Wallis test
bray_stat_KW <- kruskal.test(bray_df_wistat$value ~ as.factor(bray_df_wistat$station1)) 
print(bray_stat_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
bray_stat_KW_MC <- kruskalmc(bray_df_wistat$value ~ as.factor(bray_df_wistat$station1))  ## Defaults to P < 0.05
print(bray_stat_KW_MC)

# Test to find the letters to represent significance in a plot
bray_stat_KW_test <- bray_stat_KW_MC$dif.com$difference # select logical vector
names(bray_stat_KW_test) <- row.names(bray_stat_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
bray_stat_KW_letters <- multcompLetters(bray_stat_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
bray_stat_KW_sigs_dataframe <-  data.frame(as.vector(names(bray_stat_KW_letters$Letters)), as.vector(bray_stat_KW_letters$Letters))
colnames(bray_stat_KW_sigs_dataframe) <- c("station1", "siglabel")
bray_stat_KW_try <- merge(bray_df_wistat, bray_stat_KW_sigs_dataframe, by = "station1")

bray_stat_KW_try$station1 <-factor(bray_stat_KW_try$station1,levels=c( "MLB", "MM15","MM110"))

bray_stat_KW_sigs <- bray_stat_KW_try %>%
  select(station1, station1, siglabel) %>%
  distinct()

#ordering levels
bray_df_wistat$compfrac <- factor(bray_df_wistat$compfrac,levels=c("Sp_FL","Su_FL","Fa_FL","Sp_PA","Su_PA","Fa_PA","Sp_FLvsPA","Su_FLvsPA","Fa_FLvsPA","Sp-Su_FL","Sp-Fa_FL","Su-Fa_FL","Sp-Su_PA","Sp-Fa_PA","Su-Fa_PA","Sp-Su_FLvsPA","Sp-Fa_FLvsPA","Su-Fa_FLvsPA"))
bray_df_wistat$compdepth <- factor(bray_df_wistat$compdepth,levels=c("Sp-Su_Surf","Sp-Fa_Surf","Su-Fa_Surf","Sp-Su_Deep","Sp-Fa_Deep","Su-Fa_Deep","Sp-Su_SurfvsDeep","Sp-Fa_SurfvsDeep","Su-Fa_SurfvsDeep"))
bray_df_wistat$station1 <- factor(bray_df_wistat$station1,levels=c("MLB","MM15","MM110"))

#determine number of samples in comparisons
nrow(subset(bray_df_wistat, station1 == "MLB"))
nrow(subset(bray_df_wistat, station1 == "MM15"))
nrow(subset(bray_df_wistat, station1 == "MM110"))


#plot
pdf(file="Figures/Rplot_BetaDivStations_boxplot_FL.pdf", width= 3.5, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
bray_wistat_boxplot_FL <- ggplot(bray_df_wistat, aes(x = station1, y = value))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=compdepth),size=1,alpha=1, position = position_dodge(0.8)) +
  geom_text(data = bray_stat_KW_sigs, aes(label = siglabel, x = station1, y = 0), size = 2) +
  xlab("") + ylab("Bray-Curtis dissimilarity") + theme_bw() + ylim(0,1) +
  scale_x_discrete(breaks=c("MLB", "MM15","MM110"),labels=c("MLB\n(n=)","M15\n(n=)","M110\n(n=)")) + 
  guides(color=guide_legend(title="comparison")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 6, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 4),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); bray_wistat_boxplot_FL #top, right, bottom, left
dev.off()
```

```{r plot station comparisons PA fraction}
#only retain BC dissimilarities within each fraction, not betwee the two fractions
bray_df_wistat <- subset(bray_df, station1 == station2)
bray_df_wistat <- subset(bray_df_wistat, fraction1 == fraction2)
bray_df_wistat <- subset(bray_df_wistat, season1 != season2)
bray_df_wistat <- subset(bray_df_wistat, fraction1 == "E")

#Kruskall-Wallis test
bray_stat_KW <- kruskal.test(bray_df_wistat$value ~ as.factor(bray_df_wistat$station1)) 
print(bray_stat_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
bray_stat_KW_MC <- kruskalmc(bray_df_wistat$value ~ as.factor(bray_df_wistat$station1))  ## Defaults to P < 0.05
print(bray_stat_KW_MC)

# Test to find the letters to represent significance in a plot
bray_stat_KW_test <- bray_stat_KW_MC$dif.com$difference # select logical vector
names(bray_stat_KW_test) <- row.names(bray_stat_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
bray_stat_KW_letters <- multcompLetters(bray_stat_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
bray_stat_KW_sigs_dataframe <-  data.frame(as.vector(names(bray_stat_KW_letters$Letters)), as.vector(bray_stat_KW_letters$Letters))
colnames(bray_stat_KW_sigs_dataframe) <- c("station1", "siglabel")
bray_stat_KW_try <- merge(bray_df_wistat, bray_stat_KW_sigs_dataframe, by = "station1")

bray_stat_KW_try$station1 <-factor(bray_stat_KW_try$station1,levels=c( "MLB", "MM15","MM110"))

bray_stat_KW_sigs <- bray_stat_KW_try %>%
  select(station1, station1, siglabel) %>%
  distinct()

#ordering levels
bray_df_wistat$compfrac <- factor(bray_df_wistat$compfrac,levels=c("Sp_FL","Su_FL","Fa_FL","Sp_PA","Su_PA","Fa_PA","Sp_FLvsPA","Su_FLvsPA","Fa_FLvsPA","Sp-Su_FL","Sp-Fa_FL","Su-Fa_FL","Sp-Su_PA","Sp-Fa_PA","Su-Fa_PA","Sp-Su_FLvsPA","Sp-Fa_FLvsPA","Su-Fa_FLvsPA"))
bray_df_wistat$compdepth <- factor(bray_df_wistat$compdepth,levels=c("Sp-Su_Surf","Sp-Fa_Surf","Su-Fa_Surf","Sp-Su_Deep","Sp-Fa_Deep","Su-Fa_Deep","Sp-Su_SurfvsDeep","Sp-Fa_SurfvsDeep","Su-Fa_SurfvsDeep"))
bray_df_wistat$station1 <- factor(bray_df_wistat$station1,levels=c("MLB","MM15","MM110"))

#determine number of samples in comparisons
nrow(subset(bray_df_wistat, station1 == "MLB"))
nrow(subset(bray_df_wistat, station1 == "MM15"))
nrow(subset(bray_df_wistat, station1 == "MM110"))


#plot
pdf(file="Figures/Rplot_BetaDivStations_boxplot_PA.pdf", width= 3.5, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
bray_wistat_boxplot_PA <- ggplot(bray_df_wistat, aes(x = station1, y = value))  +
  geom_boxplot(outlier.size = 0)  + ggtitle("") +
  geom_point(aes(color=compdepth),size=1,alpha=1, position = position_dodge(0.8)) +
  geom_text(data = bray_stat_KW_sigs, aes(label = siglabel, x = station1, y = 0), size = 2) +
  xlab("") + ylab("Bray-Curtis dissimilarity") + theme_bw() + ylim(0,1) +
  scale_x_discrete(breaks=c("MLB", "MM15","MM110"),labels=c("MLB\n(n=)","M15\n(n=)","M110\n(n=)")) + 
  guides(color=guide_legend(title="comparison")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 0.5, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 6, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 4),  #Set the legend text
                                 legend.position="top", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); bray_wistat_boxplot_PA #top, right, bottom, left
dev.off()
```

```{r plot interstation comparisons both fraction, surface only}
#only retain BC dissimilarities between stations at same time, depth and fraction
bray_df_betweenstat <- subset(bray_df, station1 != station2)
bray_df_betweenstat <- subset(bray_df_betweenstat, fraction1 == fraction2)
bray_df_betweenstat <- subset(bray_df_betweenstat, season1 == season2)
bray_df_betweenstat <- subset(bray_df_betweenstat, depth1 == depth2)
bray_df_betweenstat <- subset(bray_df_betweenstat, depth1 == "S")

#Kruskall-Wallis test
bray_stat_KW <- kruskal.test(bray_df_betweenstat$value ~ as.factor(bray_df_betweenstat$compstat)) 
print(bray_stat_KW)  # show Kruskal Wallis result

# Which samples are significantly different from each other?  
bray_stat_KW_MC <- kruskalmc(bray_df_betweenstat$value ~ as.factor(bray_df_betweenstat$compstat))  ## Defaults to P < 0.05
print(bray_stat_KW_MC)

# Test to find the letters to represent significance in a plot
bray_stat_KW_test <- bray_stat_KW_MC$dif.com$difference # select logical vector
names(bray_stat_KW_test) <- row.names(bray_stat_KW_MC$dif.com) # add comparison names

# create a list with "homogenous groups" coded by letter
bray_stat_KW_letters <- multcompLetters(bray_stat_KW_test, compare="<", threshold=0.05, 
                                Letters=c(letters, LETTERS, "."), reversed = FALSE)

#  Extract the values from the multcompLetters object
bray_stat_KW_sigs_dataframe <-  data.frame(as.vector(names(bray_stat_KW_letters$Letters)), as.vector(bray_stat_KW_letters$Letters))
colnames(bray_stat_KW_sigs_dataframe) <- c("compstat", "siglabel")
bray_stat_KW_try <- merge(bray_df_betweenstat, bray_stat_KW_sigs_dataframe, by = "compstat")


bray_stat_KW_try$compstat <-factor(bray_stat_KW_try$compstat,levels=c("Sp_MLBvsMM15","Sp_MLBvsMM110","Sp_MM15vsMM110","Su_MLBvsMM15","Su_MLBvsMM110", "Su_MM15vsMM110", "Fa_MLBvsMM15","Fa_MLBvsMM110","Fa_MM15vsMM110" ))

bray_stat_KW_sigs <- bray_stat_KW_try %>%
  select(compstat, compstat, siglabel) %>%
  distinct()

#ordering levels
#bray_df_betweenstat$compfrac <- factor(bray_df_betweenstat$compfrac,levels=c("Sp_FL","Su_FL","Fa_FL","Sp_PA","Su_PA","Fa_PA","Sp_FLvsPA","Su_FLvsPA","Fa_FLvsPA","Sp-Su_FL","Sp-Fa_FL","Su-Fa_FL","Sp-Su_PA","Sp-Fa_PA","Su-Fa_PA","Sp-Su_FLvsPA","Sp-Fa_FLvsPA","Su-Fa_FLvsPA"))
#bray_df_betweenstat$compdepth <- factor(bray_df_betweenstat$compdepth,levels=c("Sp_Surf","Su_Surf","Fa_Surf","Sp_Deep","Su_Deep","Fa_Deep","Sp_SurfvsDeep","Su_SurfvsDeep","Fa_SurfvsDeep"))
bray_df_betweenstat$compstat <-factor(bray_df_betweenstat$compstat,levels=c("Sp_MLBvsMM15","Sp_MLBvsMM110","Sp_MM15vsMM110","Su_MLBvsMM15","Su_MLBvsMM110", "Su_MM15vsMM110", "Fa_MLBvsMM15","Fa_MLBvsMM110","Fa_MM15vsMM110" ))

unique(bray_df_betweenstat$compdepth)
#determine number of samples in comparisons
#nrow(subset(bray_df_betweenstat, station1 == "MLB"))
#nrow(subset(bray_df_betweenstat, station1 == "MM15"))
#nrow(subset(bray_df_betweenstat, station1 == "MM110"))

#plot
pdf(file="Figures/Rplot_BetaDiv_BetweenStations_boxplot.pdf", width= 3.5, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
bray_betweenstat_boxplot <- ggplot(bray_df_betweenstat, aes(x = compstat, y = value))  +
  geom_boxplot(outlier.size = 1)  + ggtitle("") +
 # geom_point(aes(color=fraction1),size=1,alpha=1, position = position_dodge(0.8)) +
  geom_text(data = bray_stat_KW_sigs, aes(label = siglabel, x = compstat, y = 0), size = 2) +
  xlab("") + ylab("Bray-Curtis dissimilarity") + theme_bw() + ylim(0,1) +
  scale_x_discrete(breaks=c("Sp_MLBvsMM15","Sp_MLBvsMM110","Sp_MM15vsMM110","Su_MLBvsMM15","Su_MLBvsMM110", "Su_MM15vsMM110", "Fa_MLBvsMM15","Fa_MLBvsMM110","Fa_MM15vsMM110"),labels=c("Spring\nMLB-M15","MLB-M110","M15-M110","Summer\nMLB-M15","MLB-M110", "M15-M110", "Fall\nMLB-M15","MLB-M110","M15-M110")) + 
  guides(color=guide_legend(title="comparison")) +
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=45, colour = "black", 
                                                            vjust=1, hjust = 1, size=5),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 6, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 4),  #Set the legend text
                                 legend.position="right", #Default the legend position to the right
                                 plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")); bray_betweenstat_boxplot #top, right, bottom, left
dev.off()
```


```{r final figure 4}
library(grid)
library(gridExtra)

pdf(file="Figures/Rplot_BetaDiv_boxplot_betweenstat.pdf", width= 3.5, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(bray_betweenstat_boxplot,ncol=1)
dev.off()

```

## Figure 5 and 6: diff abundant otus visualized by functional groups
```{r Deseq analysis and plots by functional group PA vs FL}
mean(sample_sums(physeq_dupmerge))
# Keep OTUs above average relative abundance of 330 reads (0.1% before scaling) for use with deseq 
a <- apply(otu_table(physeq_dupmerge), MARGIN = 1, function(x) {mean(x)} )
keep <- a > 330
physeq_dupmerge.prune <- prune_taxa(keep, physeq_dupmerge)
ntaxa(physeq_dupmerge.prune) 

write.table(tax_table(physeq_dupmerge.prune), "Figures/deseq_prune.taxtable.tsv", row.names = TRUE)

# now add functional group information from manually curated file FunGrouptoptaxa.txt
fungroupfile <- read.table("Input_data/FunGrouptoptaxa.txt", header = TRUE)
row.names(fungroupfile) <- fungroupfile$otu
#order ascending based on otu number
fungroupfile <- fungroupfile[order(fungroupfile[,1]),]

##
theme_deseq <- theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_blank(),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the x-axis labels
                                 axis.text.y = element_blank(),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position = "none",
                                 panel.grid.major.y = element_blank(),
                                 panel.grid.major.x = element_blank())
                       
##FL vs PA
# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.frac.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune, ~Season+Depth_C+Lake+Fraction)
physeq_dupmerge.prune.frac.deseq = DESeq(physeq_dupmerge.prune.frac.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.frac.deseq, cooksCutoff = FALSE,contrast=c("Fraction","B","E")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.FLvsPA <-
  res %>%
    filter(padj < alpha) 
sigtab.FLvsPA <- cbind(sigtab.FLvsPA, 
  as(tax_table(physeq_dupmerge.prune)[sigtab.FLvsPA$OTU, ], "matrix"),fungroupfile[sigtab.FLvsPA$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue <- data.frame(data.frame(tax_table(physeq_dupmerge))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge)/sum(sample_sums(physeq_dupmerge))*100)))
colnames(plotvalue) <- c("species","abundance")
row.names(plotvalue) <- plotvalue$species
plotvalue <- plotvalue[row.names(plotvalue) %in% sigtab.FLvsPA$Species, ]
sigtab.FLvsPA <- cbind(sigtab.FLvsPA,plotvalue[sigtab.FLvsPA$OTU, ])
sigtab.FLvsPA$sig <- as.factor((sigtab.FLvsPA$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.FLvsPA <- arrange(sigtab.FLvsPA, energy_short)
sigtab.FLvsPA$Genus <- factor(sigtab.FLvsPA$Genus, 
  levels = unique(sigtab.FLvsPA$Genus))
sigtab.FLvsPA$energy_short <- factor(sigtab.FLvsPA$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

#set outlier values >5 and <-10 to 5 and -10
sigtab.FLvsPA[3] <- lapply(sigtab.FLvsPA[3], function(x) ifelse(x>4.99, 4.99, x))
sigtab.FLvsPA[3] <- lapply(sigtab.FLvsPA[3], function(x) ifelse(x<(-9.99), -9.99, x))

# Plot
plot.FLvsPA <- ggplot(sigtab.FLvsPA, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype="dotted") +
  xlab("log2fold FL:PA") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.FLvsPA

##2b PAvsFL with legend
plot.FLvsPA.wlegend <- ggplot(sigtab.FLvsPA, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype="dotted") +
  xlab("log2fold FL:PA") + 
  scale_y_discrete() +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
 theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_blank(),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(size = 7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position = "right",
                                 panel.grid.major.y = element_blank(),
                                 panel.grid.major.x = element_blank()) 
plot.FLvsPA.wlegend

# Plot legend only
tmp <- ggplot_gtable(ggplot_build(plot.FLvsPA.wlegend)) 
leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
plot.legend <- tmp$grobs[[leg]] 

# Plot Y-axis only
library(gtable)
g <- ggplotGrob(plot.FLvsPA.wlegend)
plot.y <- gtable_filter(g, 'axis-l|ylab', trim=F)  # use trim depending on need

```

``` {r plots by phylum/class all data combined only for PA vs FL}

theme_deseq2 <- theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(size = 7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(size = 8),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position = "none",
                                 panel.grid.major.y = element_blank(colour="gray70", linetype = "dash"),
                                 panel.grid.major.x = element_blank())

## FL vs PA
# Change genus levels so organized by phylum
sigtab.phyl.FLvsPA <- arrange(sigtab.FLvsPA, Phylum)
sigtab.phyl.FLvsPA$Phylum <-factor(sigtab.phyl.FLvsPA$Phylum,levels=c("unclassified","Gemmatimonadetes","Chlorobi","Armatimonadetes","Chloroflexi","Planctomycetes","Verrucomicrobia","Cyanobacteria","Actinobacteria","Sphingobacteria","Cytophagia","Flavobacteria","Bacteroidetes","Deltaproteobacteria","Gammaproteobacteria","Betaproteobacteria","Alphaproteobacteria","Proteobacteria"))
sigtab.phyl.FLvsPA$energy_short <- factor(sigtab.phyl.FLvsPA$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))
for(i in 1:length(sigtab.phyl.FLvsPA$C_source)){
  sigtab.phyl.FLvsPA$C_source_sig[i]<-paste(as.character(sigtab.phyl.FLvsPA$C_source[i]),
                                  as.character(sigtab.phyl.FLvsPA$sig[i]))}
#set outlier values >5 and <-10 to 5 and -10
sigtab.phyl.FLvsPA[3] <- lapply(sigtab.phyl.FLvsPA[3], function(x) ifelse(x>4.99, 4.99, x))
sigtab.phyl.FLvsPA[3] <- lapply(sigtab.phyl.FLvsPA[3], function(x) ifelse(x<(-9.99), -9.99, x))

# Plot
plot.phylFLvsPA <- ggplot(sigtab.phyl.FLvsPA, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),colour="grey", linetype = "dotted") +
  xlab("log2fold FL:PA") + 
  scale_y_discrete(breaks=NULL) +
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) +
  theme_deseq2
plot.phylFLvsPA

# Plot with legend
pdf(file="Figures/Rplot_deseq_fungroups_phylum_PAvsFL_wlegend.pdf", width= 3.5, height=10, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
plot.phylFLvsPA_wlegend <- ggplot(sigtab.phyl.FLvsPA, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
#  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),colour="grey", linetype = "dotted") +
  xlab("log2fold Surface:Deep") + 
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7)  + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_text(face="bold", size=7),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position = "right",
                                 panel.grid.major.y = element_blank(),
                                 panel.grid.major.x = element_blank() 
  ); plot.phylFLvsPA_wlegend
dev.off()

# Plot legend only
tmp <- ggplot_gtable(ggplot_build(plot.phylFLvsPA_wlegend)) 
leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
plot.phyllegend <- tmp$grobs[[leg]] 

```

```{r final figure 5 PA vs FL betadiv + DEseq and Fig. S3}
library(grid)
library(gridExtra)


pdf(file="Figures/Rplot_BetaDiv_boxplot_PAvsFL+Deseq_combined.pdf", width= 7, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(bray_wifrac_boxplot,plot.FLvsPA.wlegend,ncol=2)
dev.off()

pdf(file="Figures/Rplot_DeseqPAvsFL_phylum_combined.pdf", width= 7, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(plot.phylFLvsPA,plot.phylFLvsPA_wlegend,ncol=2)
dev.off()

pdf(file="Figures/Rplot_fctgroups_Yaxis.pdf", width= 3.5, height=3.5, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(plot.FLvsPA.wlegend,ncol=1)
dev.off()

```

```{r Deseq analysis and plots by functional group FL only}
# remove PA samples 
physeq_dupmerge.FL <- subset_samples(physeq_dupmerge,Fraction=="B") 
physeq_dupmerge.prune.FL <- subset_samples(physeq_dupmerge.prune,Fraction=="B") 

# now add functional group information from manually curated file FunGrouptoptaxa.txt
##1. LM vs MLB
# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for season and Depth
physeq_dupmerge.prune.FL.LMvsMLB.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.FL, ~Season+Depth_C+Lake)
physeq_dupmerge.prune.FL.LMvsMLB.deseq = DESeq(physeq_dupmerge.prune.FL.LMvsMLB.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.FL.LMvsMLB.deseq, cooksCutoff = FALSE,contrast=c("Lake","MM","MLB")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.FL.LMvsMLB <-
  res %>%
    filter(padj < alpha) 
sigtab.FL.LMvsMLB <- cbind(sigtab.FL.LMvsMLB, 
  as(tax_table(physeq_dupmerge.prune.FL)[sigtab.FL.LMvsMLB$OTU, ], "matrix"),fungroupfile[sigtab.FL.LMvsMLB$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.FL <- data.frame(data.frame(tax_table(physeq_dupmerge.FL))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.FL)/sum(sample_sums(physeq_dupmerge.FL))*100)))
colnames(plotvalue.FL) <- c("species","abundance")
row.names(plotvalue.FL) <- plotvalue.FL$species
plotvalue.FL <- plotvalue.FL[row.names(plotvalue.FL) %in% sigtab.FL.LMvsMLB$Species, ]
sigtab.FL.LMvsMLB <- cbind(sigtab.FL.LMvsMLB,plotvalue.FL[sigtab.FL.LMvsMLB$OTU, ])
sigtab.FL.LMvsMLB$sig <- as.factor((sigtab.FL.LMvsMLB$padj<0.01)*1)

# Change genus levels so organized by energy_short
sigtab.FL.LMvsMLB <- arrange(sigtab.FL.LMvsMLB, energy_short)
sigtab.FL.LMvsMLB$Genus <- factor(sigtab.FL.LMvsMLB$Genus, 
  levels = unique(sigtab.FL.LMvsMLB$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.FL.LMvsMLB[3] <- lapply(sigtab.FL.LMvsMLB[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.FL.LMvsMLB[3] <- lapply(sigtab.FL.LMvsMLB[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.FL.LMvsMLB$energy_short <- factor(sigtab.FL.LMvsMLB$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.LMvsMLB.FL <- ggplot(sigtab.FL.LMvsMLB, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) + 
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Lake MI : Muskegon Lake") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq
plot.LMvsMLB.FL 
#View(sigtab.FL.LMvsMLB)

##3. Sp vs Su
#subset data to exclude Fall
physeq_dupmerge.prune.FL.spsu <- subset_samples(physeq_dupmerge.prune.FL,Season!="Fa")

# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.FL.SpvsSu.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.FL.spsu, ~Depth_C+Lake+Season)
physeq_dupmerge.prune.FL.SpvsSu.deseq = DESeq(physeq_dupmerge.prune.FL.SpvsSu.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.FL.SpvsSu.deseq, cooksCutoff = FALSE,contrast=c("Season","Su","Sp")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.FL.SpvsSu <-
  res %>%
    filter(padj < alpha) 
sigtab.FL.SpvsSu <- cbind(sigtab.FL.SpvsSu, 
  as(tax_table(physeq_dupmerge.prune.FL)[sigtab.FL.SpvsSu$OTU, ], "matrix"),fungroupfile[sigtab.FL.SpvsSu$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.FL <- data.frame(data.frame(tax_table(physeq_dupmerge.FL))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.FL)/sum(sample_sums(physeq_dupmerge.FL))*100)))
colnames(plotvalue.FL) <- c("species","abundance")
row.names(plotvalue.FL) <- plotvalue.FL$species
plotvalue.FL <- plotvalue.FL[row.names(plotvalue.FL) %in% sigtab.FL.SpvsSu$Species, ]
sigtab.FL.SpvsSu <- cbind(sigtab.FL.SpvsSu,plotvalue.FL[sigtab.FL.SpvsSu$OTU, ])
sigtab.FL.SpvsSu$sig <- as.factor((sigtab.FL.SpvsSu$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.FL.SpvsSu <- arrange(sigtab.FL.SpvsSu, energy_short)
sigtab.FL.SpvsSu$Genus <- factor(sigtab.FL.SpvsSu$Genus, 
  levels = unique(sigtab.FL.SpvsSu$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.FL.SpvsSu[3] <- lapply(sigtab.FL.SpvsSu[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.FL.SpvsSu[3] <- lapply(sigtab.FL.SpvsSu[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.FL.SpvsSu$energy_short <- factor(sigtab.FL.SpvsSu$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.SuvsSp.FL <- ggplot(sigtab.FL.SpvsSu, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Summer:Spring") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.SuvsSp.FL

##4. Su vs Fa
#subset data to exclude Fall
physeq_dupmerge.prune.FL.sufa <- subset_samples(physeq_dupmerge.prune.FL,Season!="Sp")

# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.FL.SuvsFa.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.FL.sufa, ~Depth_C+Lake+Season)
physeq_dupmerge.prune.FL.SuvsFa.deseq = DESeq(physeq_dupmerge.prune.FL.SuvsFa.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.FL.SuvsFa.deseq, cooksCutoff = FALSE,contrast=c("Season","Su","Fa")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.FL.SuvsFa <-
  res %>%
    filter(padj < alpha) 
sigtab.FL.SuvsFa <- cbind(sigtab.FL.SuvsFa, 
  as(tax_table(physeq_dupmerge.prune.FL)[sigtab.FL.SuvsFa$OTU, ], "matrix"),fungroupfile[sigtab.FL.SuvsFa$OTU, ])


#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.FL <- data.frame(data.frame(tax_table(physeq_dupmerge.FL))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.FL)/sum(sample_sums(physeq_dupmerge.FL))*100)))
colnames(plotvalue.FL) <- c("species","abundance")
row.names(plotvalue.FL) <- plotvalue.FL$species
plotvalue.FL <- plotvalue.FL[row.names(plotvalue.FL) %in% sigtab.FL.SuvsFa$Species, ]
sigtab.FL.SuvsFa <- cbind(sigtab.FL.SuvsFa,plotvalue.FL[sigtab.FL.SuvsFa$OTU, ])
sigtab.FL.SuvsFa$sig <- as.factor((sigtab.FL.SuvsFa$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.FL.SuvsFa <- arrange(sigtab.FL.SuvsFa, energy_short)
sigtab.FL.SuvsFa$Genus <- factor(sigtab.FL.SuvsFa$Genus, 
  levels = unique(sigtab.FL.SuvsFa$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.FL.SuvsFa[3] <- lapply(sigtab.FL.SuvsFa[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.FL.SuvsFa[3] <- lapply(sigtab.FL.SuvsFa[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.FL.SuvsFa$energy_short <- factor(sigtab.FL.SuvsFa$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.SuvsFa.FL <- ggplot(sigtab.FL.SuvsFa, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Summer:Fall") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.SuvsFa.FL


##5. S vs D 
#subset data to exclude mid depth 
physeq_dupmerge.prune.FL.SvsD <- subset_samples(physeq_dupmerge.prune.FL,Depth_C!="M")

# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.FL.SvsD.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.FL.SvsD, ~Lake+Season+Depth_C)
physeq_dupmerge.prune.FL.SvsD.deseq = DESeq(physeq_dupmerge.prune.FL.SvsD.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.FL.SvsD.deseq, cooksCutoff = FALSE,contrast=c("Depth_C","S","D")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.FL.SvsD <-
  res %>%
    filter(padj < alpha) 
sigtab.FL.SvsD <- cbind(sigtab.FL.SvsD, 
  as(tax_table(physeq_dupmerge.prune.FL)[sigtab.FL.SvsD$OTU, ], "matrix"),fungroupfile[sigtab.FL.SvsD$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.FL <- data.frame(data.frame(tax_table(physeq_dupmerge.FL))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.FL)/sum(sample_sums(physeq_dupmerge.FL))*100)))
colnames(plotvalue.FL) <- c("species","abundance")
row.names(plotvalue.FL) <- plotvalue.FL$species
plotvalue.FL <- plotvalue.FL[row.names(plotvalue.FL) %in% sigtab.FL.SvsD$Species, ]
sigtab.FL.SvsD <- cbind(sigtab.FL.SvsD,plotvalue.FL[sigtab.FL.SvsD$OTU, ])
sigtab.FL.SvsD$sig <- as.factor((sigtab.FL.SvsD$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.FL.SvsD <- arrange(sigtab.FL.SvsD, energy_short)
sigtab.FL.SvsD$Genus <- factor(sigtab.FL.SvsD$Genus, 
  levels = unique(sigtab.FL.SvsD$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.FL.SvsD[3] <- lapply(sigtab.FL.SvsD[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.FL.SvsD[3] <- lapply(sigtab.FL.SvsD[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.FL.SvsD$energy_short <- factor(sigtab.FL.SvsD$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.SvsD.FL <- ggplot(sigtab.FL.SvsD, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Surface:Deep") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.SvsD.FL
```

```{r Deseq analysis and plots by functional group PA only}
# remove FL samples 
physeq_dupmerge.PA <- subset_samples(physeq_dupmerge,Fraction=="E") 
physeq_dupmerge.prune.PA <- subset_samples(physeq_dupmerge.prune,Fraction=="E") 

# now add functional group information from manually curated file FunGrouptoptaxa.txt
##1. LM vs MLB
# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for season and Depth
physeq_dupmerge.prune.PA.LMvsMLB.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.PA, ~Season+Depth_C+Lake)
physeq_dupmerge.prune.PA.LMvsMLB.deseq = DESeq(physeq_dupmerge.prune.PA.LMvsMLB.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.PA.LMvsMLB.deseq, cooksCutoff = FALSE,contrast=c("Lake","MM","MLB")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.PA.LMvsMLB <-
  res %>%
    filter(padj < alpha) 
sigtab.PA.LMvsMLB <- cbind(sigtab.PA.LMvsMLB, 
  as(tax_table(physeq_dupmerge.prune.PA)[sigtab.PA.LMvsMLB$OTU, ], "matrix"),fungroupfile[sigtab.PA.LMvsMLB$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.PA <- data.frame(data.frame(tax_table(physeq_dupmerge.PA))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.PA)/sum(sample_sums(physeq_dupmerge.PA))*100)))
colnames(plotvalue.PA) <- c("species","abundance")
row.names(plotvalue.PA) <- plotvalue.PA$species
plotvalue.PA <- plotvalue.PA[row.names(plotvalue.PA) %in% sigtab.PA.LMvsMLB$Species, ]
sigtab.PA.LMvsMLB <- cbind(sigtab.PA.LMvsMLB,plotvalue.PA[sigtab.PA.LMvsMLB$OTU, ])
sigtab.PA.LMvsMLB$sig <- as.factor((sigtab.PA.LMvsMLB$padj<0.01)*1)

# Change genus levels so organized by energy_short
sigtab.PA.LMvsMLB <- arrange(sigtab.PA.LMvsMLB, energy_short)
sigtab.PA.LMvsMLB$Genus <- factor(sigtab.PA.LMvsMLB$Genus, 
  levels = unique(sigtab.PA.LMvsMLB$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.PA.LMvsMLB[3] <- lapply(sigtab.PA.LMvsMLB[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.PA.LMvsMLB[3] <- lapply(sigtab.PA.LMvsMLB[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.PA.LMvsMLB$energy_short <- factor(sigtab.PA.LMvsMLB$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.LMvsMLB.PA <- ggplot(sigtab.PA.LMvsMLB, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) + 
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Lake MI : Muskegon Lake") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq
plot.LMvsMLB.PA 
#View(sigtab.PA.LMvsMLB)

##3. Sp vs Su
#subset data to exclude Fall
physeq_dupmerge.prune.PA.spsu <- subset_samples(physeq_dupmerge.prune.PA,Season!="Fa")

# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.PA.SpvsSu.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.PA.spsu, ~Depth_C+Lake+Season)
physeq_dupmerge.prune.PA.SpvsSu.deseq = DESeq(physeq_dupmerge.prune.PA.SpvsSu.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.PA.SpvsSu.deseq, cooksCutoff = FALSE,contrast=c("Season","Su","Sp")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.PA.SpvsSu <-
  res %>%
    filter(padj < alpha) 
sigtab.PA.SpvsSu <- cbind(sigtab.PA.SpvsSu, 
  as(tax_table(physeq_dupmerge.prune.PA)[sigtab.PA.SpvsSu$OTU, ], "matrix"),fungroupfile[sigtab.PA.SpvsSu$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.PA <- data.frame(data.frame(tax_table(physeq_dupmerge.PA))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.PA)/sum(sample_sums(physeq_dupmerge.PA))*100)))
colnames(plotvalue.PA) <- c("species","abundance")
row.names(plotvalue.PA) <- plotvalue.PA$species
plotvalue.PA <- plotvalue.PA[row.names(plotvalue.PA) %in% sigtab.PA.SpvsSu$Species, ]
sigtab.PA.SpvsSu <- cbind(sigtab.PA.SpvsSu,plotvalue.PA[sigtab.PA.SpvsSu$OTU, ])
sigtab.PA.SpvsSu$sig <- as.factor((sigtab.PA.SpvsSu$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.PA.SpvsSu <- arrange(sigtab.PA.SpvsSu, energy_short)
sigtab.PA.SpvsSu$Genus <- factor(sigtab.PA.SpvsSu$Genus, 
  levels = unique(sigtab.PA.SpvsSu$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.PA.SpvsSu[3] <- lapply(sigtab.PA.SpvsSu[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.PA.SpvsSu[3] <- lapply(sigtab.PA.SpvsSu[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.PA.SpvsSu$energy_short <- factor(sigtab.PA.SpvsSu$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.SuvsSp.PA <- ggplot(sigtab.PA.SpvsSu, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Summer:Spring") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.SuvsSp.PA

##4. Su vs Fa
#subset data to exclude Fall
physeq_dupmerge.prune.PA.sufa <- subset_samples(physeq_dupmerge.prune.PA,Season!="Sp")

# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.PA.SuvsFa.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.PA.sufa, ~Depth_C+Lake+Season)
physeq_dupmerge.prune.PA.SuvsFa.deseq = DESeq(physeq_dupmerge.prune.PA.SuvsFa.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.PA.SuvsFa.deseq, cooksCutoff = FALSE,contrast=c("Season","Su","Fa")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.PA.SuvsFa <-
  res %>%
    filter(padj < alpha) 
sigtab.PA.SuvsFa <- cbind(sigtab.PA.SuvsFa, 
  as(tax_table(physeq_dupmerge.prune.PA)[sigtab.PA.SuvsFa$OTU, ], "matrix"),fungroupfile[sigtab.PA.SuvsFa$OTU, ])


#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.PA <- data.frame(data.frame(tax_table(physeq_dupmerge.PA))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.PA)/sum(sample_sums(physeq_dupmerge.PA))*100)))
colnames(plotvalue.PA) <- c("species","abundance")
row.names(plotvalue.PA) <- plotvalue.PA$species
plotvalue.PA <- plotvalue.PA[row.names(plotvalue.PA) %in% sigtab.PA.SuvsFa$Species, ]
sigtab.PA.SuvsFa <- cbind(sigtab.PA.SuvsFa,plotvalue.PA[sigtab.PA.SuvsFa$OTU, ])
sigtab.PA.SuvsFa$sig <- as.factor((sigtab.PA.SuvsFa$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.PA.SuvsFa <- arrange(sigtab.PA.SuvsFa, energy_short)
sigtab.PA.SuvsFa$Genus <- factor(sigtab.PA.SuvsFa$Genus, 
  levels = unique(sigtab.PA.SuvsFa$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.PA.SuvsFa[3] <- lapply(sigtab.PA.SuvsFa[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.PA.SuvsFa[3] <- lapply(sigtab.PA.SuvsFa[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.PA.SuvsFa$energy_short <- factor(sigtab.PA.SuvsFa$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.SuvsFa.PA <- ggplot(sigtab.PA.SuvsFa, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Summer:Fall") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.SuvsFa.PA


##5. S vs D 
#subset data to exclude mid depth 
physeq_dupmerge.prune.PA.SvsD <- subset_samples(physeq_dupmerge.prune.PA,Depth_C!="M")

# Convert to deseq object, comparing Muskegon Lake vs Lake Michigan, controlling for sason, fraction and Depth
physeq_dupmerge.prune.PA.SvsD.deseq <- phyloseq_to_deseq2(physeq_dupmerge.prune.PA.SvsD, ~Lake+Season+Depth_C)
physeq_dupmerge.prune.PA.SvsD.deseq = DESeq(physeq_dupmerge.prune.PA.SvsD.deseq, test = "Wald", fitType = "parametric", betaPrior=FALSE)

# Summarize results and combine with tax_table and fungroup dataframes
res = data.frame(results(physeq_dupmerge.prune.PA.SvsD.deseq, cooksCutoff = FALSE,contrast=c("Depth_C","S","D")))
res = data.frame(OTU = row.names(res), res)
alpha = 1
sigtab.PA.SvsD <-
  res %>%
    filter(padj < alpha) 
sigtab.PA.SvsD <- cbind(sigtab.PA.SvsD, 
  as(tax_table(physeq_dupmerge.prune.PA)[sigtab.PA.SvsD$OTU, ], "matrix"),fungroupfile[sigtab.PA.SvsD$OTU, ])

#calculate average abundance acorss all datasets to scale geom_point in plot
plotvalue.PA <- data.frame(data.frame(tax_table(physeq_dupmerge.PA))$Species,as.vector(as.numeric(taxa_sums(physeq_dupmerge.PA)/sum(sample_sums(physeq_dupmerge.PA))*100)))
colnames(plotvalue.PA) <- c("species","abundance")
row.names(plotvalue.PA) <- plotvalue.PA$species
plotvalue.PA <- plotvalue.PA[row.names(plotvalue.PA) %in% sigtab.PA.SvsD$Species, ]
sigtab.PA.SvsD <- cbind(sigtab.PA.SvsD,plotvalue.PA[sigtab.PA.SvsD$OTU, ])
sigtab.PA.SvsD$sig <- as.factor((sigtab.PA.SvsD$padj<0.01)*1)

# Change genus levels so organized by phylum
sigtab.PA.SvsD <- arrange(sigtab.PA.SvsD, energy_short)
sigtab.PA.SvsD$Genus <- factor(sigtab.PA.SvsD$Genus, 
  levels = unique(sigtab.PA.SvsD$Genus))
#set logfoldchange values > 10 to 10 and < -10 to -10
sigtab.PA.SvsD[3] <- lapply(sigtab.PA.SvsD[3], function(x) ifelse(x>9.99, 9.99, x))
sigtab.PA.SvsD[3] <- lapply(sigtab.PA.SvsD[3], function(x) ifelse(x<(-9.99), -9.99, x))

sigtab.PA.SvsD$energy_short <- factor(sigtab.PA.SvsD$energy_short,levels=c("org/litho(S,NH4+)/photo(BCHL)","org/litho(S)/photo(BCHL)","org/photo(BCHL)","org/photo(BR)","litho(NH4+)","org(C1)","org","photo(CHL)"))

# Plot
plot.SvsD.PA <- ggplot(sigtab.PA.SvsD, 
  aes(y = energy_short, x = log2FoldChange, color = C_source, shape = sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5),colour="grey",linetype = "dotted") +
  xlab("log2fold Surface:Deep") + 
  scale_y_discrete(breaks=NULL) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) + 
  scale_shape_manual(name = "p-value", breaks = c("0", "1"), 
                    labels = c("p>0.01", "p<0.01"),
                    values = c("0" = 21, "1"= 19)) +
  theme_deseq 
plot.SvsD.PA
```

``` {r plots by phylum/class FL only}
##1. LM vs MLB
## order phyla/classes for display (~by abundance of phyla)
# Change genus levels so organized by phylum
sigtab.phyl.LMvsMLB <- arrange(sigtab.LMvsMLB, Phylum)
sigtab.phyl.LMvsMLB$Phylum <-factor(sigtab.phyl.LMvsMLB$Phylum,levels=c("unclassified","Gemmatimonadetes","Chlorobi","Armatimonadetes","Chloroflexi","Planctomycetes","Verrucomicrobia","Cyanobacteria","Actinobacteria","Sphingobacteria","Cytophagia","Flavobacteria","Bacteroidetes","Deltaproteobacteria","Gammaproteobacteria","Betaproteobacteria","Alphaproteobacteria","Proteobacteria"))
for(i in 1:length(sigtab.phyl.LMvsMLB$C_source)){
  sigtab.phyl.LMvsMLB$C_source_sig[i]<-paste(as.character(sigtab.phyl.LMvsMLB$C_source[i]),
                                  as.character(sigtab.phyl.LMvsMLB$sig[i]))}

theme_deseq2 <- theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(size = 7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(size = 8),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position = "none",
                                 panel.grid.major.y = element_blank(colour="gray70", linetype = "dash"),
                                 panel.grid.major.x = element_blank())

# Plot
plot.phylLMvsMLB <- ggplot(sigtab.phyl.LMvsMLB, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),color="gray95") +
  xlab("log2fold Lake MI:Muskegon Lake") + 
  scale_y_discrete(breaks=NULL) +
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) +
  theme_deseq2
plot.phylLMvsMLB 
#View(sigtab.phyl.LMvsMLB)

##2. FL vs PA
# Change genus levels so organized by phylum
sigtab.phyl.FLvsPA <- arrange(sigtab.FLvsPA, Phylum)
sigtab.phyl.FLvsPA$Phylum <-factor(sigtab.phyl.FLvsPA$Phylum,levels=c("unclassified","Gemmatimonadetes","Chlorobi","Armatimonadetes","Chloroflexi","Planctomycetes","Verrucomicrobia","Cyanobacteria","Actinobacteria","Sphingobacteria","Cytophagia","Flavobacteria","Bacteroidetes","Deltaproteobacteria","Gammaproteobacteria","Betaproteobacteria","Alphaproteobacteria","Proteobacteria"))
for(i in 1:length(sigtab.phyl.FLvsPA$C_source)){
  sigtab.phyl.FLvsPA$C_source_sig[i]<-paste(as.character(sigtab.phyl.FLvsPA$C_source[i]),
                                  as.character(sigtab.phyl.FLvsPA$sig[i]))}

# Plot
plot.phylFLvsPA <- ggplot(sigtab.phyl.FLvsPA, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),color="gray95") +
  xlab("log2fold FL:PA") + 
  scale_y_discrete(breaks=NULL) +
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) +
  theme_deseq2
plot.phylFLvsPA


##3. Sp vs Su
# Change genus levels so organized by phylum
sigtab.phyl.SpvsSu <- arrange(sigtab.SpvsSu, Phylum)
sigtab.phyl.SpvsSu$Phylum <-factor(sigtab.phyl.SpvsSu$Phylum,levels=c("unclassified","Gemmatimonadetes","Chlorobi","Armatimonadetes","Chloroflexi","Planctomycetes","Verrucomicrobia","Cyanobacteria","Actinobacteria","Sphingobacteria","Cytophagia","Flavobacteria","Bacteroidetes","Deltaproteobacteria","Gammaproteobacteria","Betaproteobacteria","Alphaproteobacteria","Proteobacteria"))
for(i in 1:length(sigtab.phyl.SpvsSu$C_source)){
  sigtab.phyl.SpvsSu$C_source_sig[i]<-paste(as.character(sigtab.phyl.SpvsSu$C_source[i]),
                                  as.character(sigtab.phyl.SpvsSu$sig[i]))}


# Plot
plot.phylSuvsSp <- ggplot(sigtab.phyl.SpvsSu, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),color="gray95") +
  xlab("log2fold Summer:Spring") + 
  scale_y_discrete(breaks=NULL) +
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) +
  theme_deseq2
plot.phylSuvsSp

##4. Su vs Fa
# Change genus levels so organized by phylum
sigtab.phyl.SuvsFa <- arrange(sigtab.SuvsFa, Phylum)
sigtab.phyl.SuvsFa$Phylum <-factor(sigtab.phyl.SuvsFa$Phylum,levels=c("unclassified","Gemmatimonadetes","Chlorobi","Armatimonadetes","Chloroflexi","Planctomycetes","Verrucomicrobia","Cyanobacteria","Actinobacteria","Sphingobacteria","Cytophagia","Flavobacteria","Bacteroidetes","Deltaproteobacteria","Gammaproteobacteria","Betaproteobacteria","Alphaproteobacteria","Proteobacteria"))
for(i in 1:length(sigtab.phyl.SuvsFa$C_source)){
  sigtab.phyl.SuvsFa$C_source_sig[i]<-paste(as.character(sigtab.phyl.SuvsFa$C_source[i]),
                                  as.character(sigtab.phyl.SuvsFa$sig[i]))}

# Plot
plot.phylSuvsFa <- ggplot(sigtab.phyl.SuvsFa, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(10,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),color="gray95") +
  xlab("log2fold Summer:Fall") + 
  scale_y_discrete(breaks=NULL) +
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) +
  theme_deseq2

plot.phylSuvsFa


##5. S vs D 
# Change genus levels so organized by phylum
sigtab.phyl.SvsD <- arrange(sigtab.SvsD, Phylum)
sigtab.phyl.SvsD$Phylum <-factor(sigtab.phyl.SvsD$Phylum,levels=c("unclassified","Gemmatimonadetes","Chlorobi","Armatimonadetes","Chloroflexi","Planctomycetes","Verrucomicrobia","Cyanobacteria","Actinobacteria","Sphingobacteria","Cytophagia","Flavobacteria","Bacteroidetes","Deltaproteobacteria","Gammaproteobacteria","Betaproteobacteria","Alphaproteobacteria","Proteobacteria"))
for(i in 1:length(sigtab.phyl.SvsD$C_source)){
  sigtab.phyl.SvsD$C_source_sig[i]<-paste(as.character(sigtab.phyl.SvsD$C_source[i]),
                                  as.character(sigtab.phyl.SvsD$sig[i]))}


# Plot
plot.phylSvsD <- ggplot(sigtab.phyl.SvsD, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  geom_hline(yintercept=c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5),color="gray95") +
  xlab("log2fold Suface:Deep") + 
  scale_y_discrete(breaks=NULL) +
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7) +
  theme_deseq2

plot.phylSvsD

# Plot with legend
pdf(file="Figures/Rplot_deseq_fungroups_phylum_SvsD_wlegend.PA.pdf", width= 3.5, height=10, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
plot.phylSvsD_wlegend.PA <- ggplot(sigtab.phyl.SvsD, 
  aes(y = Phylum, x = log2FoldChange, color = energy_short, shape = C_source_sig, size = abundance)) + 
  xlim(5,-10) +
  geom_vline(xintercept = 0, colour="grey", linetype = "longdash") +
  xlab("log2fold Surface:Deep") + 
  scale_shape_manual(name = "C source and p-value", breaks = c("autotroph 0", "heterotroph 0","mixotroph 0","autotroph 1", "heterotroph 1","mixotroph 1"), 
                     labels = c("autotroph, p > 0.01", "heterotroph, p > 0.01","mixotroph, p > 0.01","autotroph, p < 0.01", "heterotroph, p < 0.01","mixotroph, p < 0.01"),
                     values = c("autotroph 0" = 1, "heterotroph 0" = 0,"mixotroph 0" = 2,"autotroph 1" = 19, "heterotroph 1" = 15,"mixotroph 1" = 17)) +
  geom_jitter(position = position_jitter(height = 0.4),alpha=0.7)  + 
  theme(plot.title = element_text(face="bold", size=7),  #Set the plot title
                                 strip.text.x = element_blank(),  #Set the facet titles on x-axis 
                                 strip.text.y = element_blank(),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=7),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=7),  #Set the y-axis title
                                 axis.text.x = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the x-axis labels
                                 axis.text.y = element_text(angle=0, colour = "black", 
                                                            vjust=1, hjust = 1, size=7),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position = "top",
                                 panel.grid.major.y = element_blank(),
                                 panel.grid.major.x = element_blank() 
  ); plot.phylSvsD_wlegend.PA
dev.off()

# Plot legend only
tmp <- ggplot_gtable(ggplot_build(plot.phylSvsD_wlegend.PA)) 
leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
plot.phyllegend <- tmp$grobs[[leg]] 

```


```{r final figure 6}
library(grid)
library(gridExtra)

pdf(file="Figures/Rplot_deseq_fungroups_noY_byfraction.pdf", width= 7, height=12, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(plot.LMvsMLB.FL,plot.LMvsMLB.PA,plot.SuvsSp.FL,plot.SuvsSp.PA,plot.SuvsFa.FL,plot.SuvsFa.PA,plot.SvsD.FL,plot.SvsD.PA,plot.legend, ncol=2)
dev.off()

#for annotating, write table to identify taxa - also added to Table S2
write.table(sigtab.phyl.FLvsPA, "Figures/sigtab.FLvsPA", row.names = TRUE)
write.table(sigtab.FL.LMvsMLB, "Figures/sigtab.FL.LMvsMLB", row.names = TRUE)
write.table(sigtab.FL.SpvsSu, "Figures/sigtab.FL.SpvsSu", row.names = TRUE)
write.table(sigtab.FL.SuvsFa, "Figures/sigtab.FL.SuvsFa", row.names = TRUE)
write.table(sigtab.FL.SvsD, "Figures/sigtab.FL.SvsD", row.names = TRUE)
write.table(sigtab.PA.LMvsMLB, "Figures/sigtab.PA.LMvsMLB", row.names = TRUE)
write.table(sigtab.PA.SpvsSu, "Figures/sigtab.PA.SpvsSu", row.names = TRUE)
write.table(sigtab.PA.SuvsFa, "Figures/sigtab.PA.SuvsFa", row.names = TRUE)
write.table(sigtab.PA.SvsD, "Figures/sigtab.PA.SvsD", row.names = TRUE)

#for number of diff ab OTUs
sum(as.numeric(as.character(sigtab.FL.LMvsMLB[which(sigtab.FL.LMvsMLB[,3]>0),22])))
sum(as.numeric(as.character(sigtab.FL.LMvsMLB[which(sigtab.FL.LMvsMLB[,3]<0),22])))
sum(as.numeric(as.character(sigtab.PA.LMvsMLB[which(sigtab.PA.LMvsMLB[,3]>0),22])))
sum(as.numeric(as.character(sigtab.PA.LMvsMLB[which(sigtab.PA.LMvsMLB[,3]<0),22])))
sum(as.numeric(as.character(sigtab.FL.SpvsSu[which(sigtab.FL.SpvsSu[,3]>0),22])))
sum(as.numeric(as.character(sigtab.FL.SpvsSu[which(sigtab.FL.SpvsSu[,3]<0),22])))
sum(as.numeric(as.character(sigtab.PA.SpvsSu[which(sigtab.PA.SpvsSu[,3]>0),22])))
sum(as.numeric(as.character(sigtab.PA.SpvsSu[which(sigtab.PA.SpvsSu[,3]<0),22])))
sum(as.numeric(as.character(sigtab.FL.SuvsFa[which(sigtab.FL.SuvsFa[,3]>0),22])))
sum(as.numeric(as.character(sigtab.FL.SuvsFa[which(sigtab.FL.SuvsFa[,3]<0),22])))
sum(as.numeric(as.character(sigtab.PA.SuvsFa[which(sigtab.PA.SuvsFa[,3]>0),22])))
sum(as.numeric(as.character(sigtab.PA.SuvsFa[which(sigtab.PA.SuvsFa[,3]<0),22])))
sum(as.numeric(as.character(sigtab.FL.SvsD[which(sigtab.FL.SvsD[,3]>0),22])))
sum(as.numeric(as.character(sigtab.FL.SvsD[which(sigtab.FL.SvsD[,3]<0),22])))
sum(as.numeric(as.character(sigtab.PA.SvsD[which(sigtab.PA.SvsD[,3]>0),22])))
sum(as.numeric(as.character(sigtab.PA.SvsD[which(sigtab.PA.SvsD[,3]<0),22])))


pdf(file="Figures/Rplot_deseq_fungroups_phylum_noY.pdf", width= 7, height=10, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(plot.phylLMvsMLB,plot.phyllegend,plot.phylSuvsSp,plot.phylSuvsFa,plot.phylFLvsPA,plot.phylSvsD, ncol=2)
dev.off()

```

```{r Figure 7: plot specific OTUs}
#Michelle's function to plot otu data
plot_otus <- function(df, otu, taxrank) {
ggplot(df,aes(x = Season, y = Abundance, group = StatFrac, color = Station, shape = Station, linetype = Fraction, fill = StatFrac)) +
geom_point(size = 2) +
geom_line(size = 0.7) +
ggtitle(paste(df[1, taxrank], otu)) +
scale_shape_manual(name = "station", breaks = c("MLB", "MM15",  "MM110"), 
                     labels = c("MLB", "M15",  "M110"),
                     values = c("MLB" = 21, "MM15" = 22, "MM110" = 24)) +
scale_color_manual(name = "station", breaks = c("MLB","MM15","MM110"), 
                     labels = c("MLB","M15","M110"),
                     values = c("MLB" = "darkgreen", "MM15" = "green", "MM110" = "blue")) +
scale_fill_manual(name = "fraction", breaks = c("MLB B","MLB E","MM15 B","MM15 E","MM110 B","MM110 E"), 
                     labels = c("MLB FL","MLB PA","MM15 FL","MM15 PA","MM110 FL","MM110 PA"),
                     values = c("MLB B" = "darkgreen", "MLB E" = "white", "MM15 B" = "green", "MM15 E" = "white", "MM110 B" = "blue", "MM110 E" = "white")) +
facet_grid(Depth_C ~.) +
ylab("rel. abund") + 
scale_x_discrete(breaks =c("Sp","Su","Fa"),drop = FALSE) +
theme(axis.text.x = element_blank(), 
  axis.text.y = element_text(size = 7),
  strip.text.y = element_blank(),  #Set the facet titles on x-axis 
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  legend.position = "none",plot.title =element_text(size = 7, face = "bold")
)
}

# Function to order sample levels correctly
order_samples <- function(df) {
  df$xvalue <-factor(df$xvalue, levels =c("Sp S","Su S","Fa S","Su M","Sp D","Su D","Fa D"))
  df$StatFrac <-factor(df$StatFrac, levels =c("MLB B","MLB E","MM15 B","MM15 E","MM110 B","MM110 E"))
  df$Season <-factor(df$Season, levels =c("Sp","Su","Fa"))
  df$Depth_C <-factor(df$Depth_C, levels =c("S","M","D"))
  return(df)
}

#prep dataframe
physeq_dupmerge.prune.noM<-subset_samples(physeq_dupmerge.prune,Depth_C!="M")
select <-physeq_dupmerge.prune.noM %>%
  transform_sample_counts(function(x) {x/sum(x)}) %>%
  subset_taxa(Species=="Otu000003"|Tribe == "LD12"|Tribe == "Lhab-A1"|Genus == "Nitrosospira"|Phylum =="Chloroflexi"|Genus == "Pnec"|Tribe == "Algor"|Species=="Otu000019"|Species=="Otu000022"|Tribe == "Algor"|Tribe == "Aquir"|Tribe == "AcI-B1"|Tribe == "LD28"|Family == "LD19"|Family == "Methylophilaceae"|Order=="Methylococcales") %>%
  psmelt()

select$xvalue <- paste(select$Season,select$Depth_C)
select$StatFrac <- paste(select$Station,select$Fraction)
select <- order_samples(select)

select_names <-as.list(levels(select$Species))
names(select_names) <-levels(select$Species)

select <- select[!(select$Station!="MLB" & select$Depth_C=="S" & select$DayNight=="N"),]

select_plots <-lapply(select_names,
  function(otu) {
    df_otu <-filter(select, OTU == otu)
    plot <- plot_otus(df = df_otu, otu = otu, taxrank = "Phylum")
    return(plot)
  }
)

blank <- grid.rect(gp=gpar(col="white"))

pdf(file="Figures/Rplot_lineplots.pdf", width= 8.5, height=12.8, pointsize= 8, colormodel="cmyk", useDingbats = FALSE)
grid.arrange(select_plots$Otu000003,select_plots$Otu000007,select_plots$Otu000008,select_plots$Otu000073,select_plots$Otu000081,select_plots$Otu000460,
             select_plots$Otu000005,select_plots$Otu000010,select_plots$Otu000019,select_plots$Otu000022,
             select_plots$Otu000119,
             select_plots$Otu000002,select_plots$Otu000021,
             select_plots$Otu000049,select_plots$Otu105848,select_plots$Otu105854,select_plots$Otu106400,
             select_plots$Otu105919,select_plots$Otu105948,select_plots$Otu105979,
             ncol = 3)
dev.off()
```

##Table 3: BIOENV analysis
```{r bioenv analysis}
physeq_dupmerge.prune.envsub <-subset_samples(physeq_dupmerge.prune,Duplicates=="Sp13.ED.MM15.SD.1"|Duplicates=="Sp13.ED.MM110.SD.1"|Duplicates=="Sp13.ED.MM110.SN.1"|Duplicates=="Sp13.ED.MM15.SN.1"|Duplicates=="Su13.ED.MM15.SN.1"|Duplicates=="Su13.ED.MM15.SD.1"|Duplicates=="Su13.ED.MM110.DCMD.1"|Duplicates=="Su13.ED.MM110.SD.1"|Duplicates=="Su13.ED.MM110.SN.1"|Duplicates=="Fa13.ED.MM15.SN.1"|Duplicates=="Fa13.ED.MM15.SD.1"|Duplicates=="Fa13.ED.MM110.DN.1"|Duplicates=="Fa13.ED.MM110.SD.1"|Duplicates=="Fa13.ED.MM110.SN.1"|Duplicates=="Sp13.BD.MM15.SD.1"|Duplicates=="Sp13.BD.MM110.SD.1"|Duplicates=="Sp13.BD.MM110.SN.1"|Duplicates=="Sp13.BD.MM15.SN.1"|Duplicates=="Su13.BD.MM15.SN.1"|Duplicates=="Su13.BD.MM15.SD.1"|Duplicates=="Su13.BD.MM110.DCMD.1"|Duplicates=="Su13.BD.MM110.DN.1"|Duplicates=="Su13.BD.MM110.SD.1"|Duplicates=="Su13.BD.MM110.SN.1"|Duplicates=="Fa13.BD.MM15.SN.1"|Duplicates=="Fa13.BD.MM15.SD.1"|Duplicates=="Fa13.BD.MM110.DN.1"|Duplicates=="Fa13.BD.MM110.SD.1"|Duplicates=="Fa13.BD.MM110.SN.1")

table_bio <- data.frame(t(otu_table(physeq_dupmerge.prune.envsub)))
table_bio_PA <- data.frame(t(otu_table(subset_samples(physeq_dupmerge.prune.envsub,Fraction=="E"))))
table_bio_FL <- data.frame(t(otu_table(subset_samples(physeq_dupmerge.prune.envsub,Fraction=="B"))))

table_env <- read.table("Input_data/bioenv_envdata.txt",sep="\t",header=T,row.names=1,as.is=T)
table_env_PA <- read.table("Input_data/bioenv_envdata_PA.txt",sep="\t",header=T,row.names=1,as.is=T)
table_env_FL <- read.table("Input_data/bioenv_envdata_FL.txt",sep="\t",header=T,row.names=1,as.is=T)

#testing normality of environmental data - log transform all with p<0.01
shapiro.test(table_env_FL$T) #p=0.04
shapiro.test(table_env_FL$DO) #p=0.01
shapiro.test(table_env_FL$Chla) #p=0.002
shapiro.test(table_env_FL$TP) #p=0.000006
shapiro.test(table_env_FL$POC) #p=0.16
shapiro.test(table_env_FL$PON) #p=0.23
shapiro.test(table_env_FL$DOC) #p=0.0001
shapiro.test(table_env_FL$SiO2) #p=0.03
shapiro.test(table_env_FL$TSS) #p=0.000006
shapiro.test(table_env_FL$PAR) #p=0.000006

#run bioenv test
bioenv_all <- bioenv(table_bio ~T+DO+log(Chla)+log(TP)+POC+PON+log(DOC)+SiO2+log(TSS)+log(PAR), table_env, method = "spearman", index = "bray", trace = TRUE, partial = NULL)
bioenv_PA <- bioenv(table_bio_PA ~T+DO+log(Chla)+log(TP)+POC+PON+log(DOC)+SiO2+log(TSS)+log(PAR), table_env_PA, method = "spearman", index = "bray", trace = TRUE, partial = NULL)
bioenv_FL <- bioenv(table_bio_FL ~T+DO+log(Chla)+log(TP)+POC+PON+log(DOC)+SiO2+log(TSS)+log(PAR), table_env_FL, method = "spearman", index = "bray", trace = TRUE, partial = NULL)

summary(bioenv_FL) # T log(Chla) SiO2 r=0.7557, best factor is Chla with r=0.64       --- p<0.01
summary(bioenv_PA) #T log(TSS) r=0.7718, best factor is T with r=0.62                 --- p<0.01
summary(bioenv_all) #T DO log(Chla) log(DOC) r=0.1421, best factor is DO with r=0.12  --- p=0.26

#testing PA with best model parameters for FL and vice versa
bioenv(table_bio_PA ~T+log(Chla)+SiO2, table_env_PA, method = "spearman", index = "bray", trace = TRUE, partial = NULL)
#T log(Chla) with r=0.69
bioenv(table_bio_PA ~log(Chla), table_env_PA, method = "spearman", index = "bray", trace = TRUE, partial = NULL)
#r = 0.38

bioenv(table_bio_FL ~T+log(TSS), table_env_FL, method = "spearman", index = "bray", trace = TRUE, partial = NULL)
#T log(TSS) with r=0.66
bioenv(table_bio_FL ~T, table_env_FL, method = "spearman", index = "bray", trace = TRUE, partial = NULL)
#r = 0.45

#performing random permutations of row names to get p value for correlation FL
x <- 0
result <- vector(mode = "numeric", length = 100)
while (x < 100){
	table_bio_rand<-table_bio[sample(1:nrow(table_bio_FL)),] 
	x <- x + 1
	bioenv_rand <- bioenv(table_bio_rand ~T+DO+log(Chla)+log(TP)+POC+PON+log(DOC)+SiO2+log(TSS)+log(PAR), table_env_FL,  method = "spearman", index = "bray", trace = TRUE, partial = NULL)
	tmp<-data.frame(summary(bioenv_rand)[2])
	result[x]=max(tmp)
	print(x)
	}
hist(result)
print(mean(result))#0.18
print(max(result))#0.55
print(quantile(result,0.95))#0.38
print(quantile(result,0.99)) #0.50

#performing random permutations of row names to get p value for correlation PA
x <- 0
result <- vector(mode = "numeric", length = 100)
while (x < 100){
	table_bio_rand<-table_bio[sample(1:nrow(table_bio_PA)),] 
	x <- x + 1
	bioenv_rand <- bioenv(table_bio_rand ~T+DO+log(Chla)+log(TP)+POC+PON+log(DOC)+SiO2+log(TSS)+log(PAR), table_env_PA,  method = "spearman", index = "bray", trace = TRUE, partial = NULL)
	tmp<-data.frame(summary(bioenv_rand)[2])
	result[x]=max(tmp)
	print(x)
	}
hist(result)
print(mean(result))#0.17
print(max(result))#0.59
print(quantile(result,0.95))#0.38
print(quantile(result,0.99)) #0.53

#performing random permutations of row names to get p value for correlation all
x <- 0
result <- vector(mode = "numeric", length = 100)
while (x < 100){
	table_bio_rand<-table_bio[sample(1:nrow(table_bio)),] 
	x <- x + 1
	bioenv_rand <- bioenv(table_bio_rand ~T+DO+log(Chla)+log(TP)+POC+PON+log(DOC)+SiO2+log(TSS)+log(PAR), table_env,  method = "spearman", index = "bray", trace = TRUE, partial = NULL)
	tmp<-data.frame(summary(bioenv_rand)[2])
	result[x]=max(tmp)
	print(x)
	}
hist(result)
print(mean(result))#0.10
print(max(result))#0.36
print(quantile(result,0.62))#0.1228
print(quantile(result,0.74))#0.1421
print(quantile(result,0.95))#0.22
print(quantile(result,0.99)) #0.24

```


